# 模型训练和评估验证

# 导入包

In [2]:
import xgboost as xgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
import pandas as pd
import numpy as np
import time
import re

# 导入特征工程后的数据

In [3]:
inputfile = '../datasets/data_feature_processed.csv'
modelfile = '../model/xgb.pkl'

In [26]:
data = pd.read_csv(inputfile)

# XGBoost模型训练

In [37]:
df = data.copy()
df.sort_values(by=['loc_id','time_stamp'],inplace=True)

In [38]:
df.head()

time_stamp  loc_id  num_of_people  hour_of_day  month_of_year  \
0  2017-04-01 00:00:00       1          172.0            0              4   
1  2017-04-01 01:00:00       1          192.0            1              4   
2  2017-04-01 02:00:00       1          190.0            2              4   
3  2017-04-01 03:00:00       1          177.0            3              4   
4  2017-04-01 04:00:00       1          216.0            4              4   

   day_of_week  week_of_year  history_1h_population  history_2h_population  \
0            5            13                  208.0                  329.0   
1            5            13                  172.0                  208.0   
2            5            13                  192.0                  172.0   
3            5            13                  190.0                  192.0   
4            5            13                  177.0                  190.0   

   history_3h_population ...  24  25  26  27  28  29  30  31  32  33  
0                  368.0 ...   0   0   0   0   0   0   0   0   0   0  
1                  329.0 ...   0   0   0   0   0   0   0   0   0   0  
2                  208.0 ...   0   0   0   0   0   0   0   0   0   0  
3                  172.0 ...   0   0   0   0   0   0   0   0   0   0  
4                  192.0 ...   0   0   0   0   0   0   0   0   0   0  

[5 rows x 93 columns]

In [39]:
df.tail()

time_stamp  loc_id  num_of_people  hour_of_day  \
163147  2017-10-31 19:00:00      33          407.0           19   
163148  2017-10-31 20:00:00      33          496.0           20   
163149  2017-10-31 21:00:00      33          434.0           21   
163150  2017-10-31 22:00:00      33          272.0           22   
163151  2017-10-31 23:00:00      33          120.0           23   

        month_of_year  day_of_week  week_of_year  history_1h_population  \
163147             10            1            44                  919.0   
163148             10            1            44                  407.0   
163149             10            1            44                  496.0   
163150             10            1            44                  434.0   
163151             10            1            44                  272.0   

        history_2h_population  history_3h_population ...  24  25  26  27  28  \
163147                 1540.0                  740.0 ...   0   0   0   0   0   
163148                  919.0                 1540.0 ...   0   0   0   0   0   
163149                  407.0                  919.0 ...   0   0   0   0   0   
163150                  496.0                  407.0 ...   0   0   0   0   0   
163151                  434.0                  496.0 ...   0   0   0   0   0   

        29  30  31  32  33  
163147   0   0   0   0   1  
163148   0   0   0   0   1  
163149   0   0   0   0   1  
163150   0   0   0   0   1  
163151   0   0   0   0   1  

[5 rows x 93 columns]

## 1. 划分训练集和测试集

In [40]:
train = df
test = df[df.time_stamp>='2017-10-01']
train_X = train.iloc[:,3:]
train_y = train['num_of_people']
test_X = test.iloc[:,3:]
test_y = test['num_of_people']
print(train_X.shape,test_X.shape)

(163152, 90) (18216, 90)


## 2. XGBoost训练

In [41]:
model_xgb = xgboost_model_train(train_X,train_y,test_X,test_y)

[0]	train-rmse:693.909	test-rmse:1039.55
Multiple eval metrics have been passed: 'test-rmse' will be used for early stopping.

Will train until test-rmse hasn't improved in 20 rounds.
[1]	train-rmse:674.878	test-rmse:1011.63
[2]	train-rmse:656.41	test-rmse:983.985
[3]	train-rmse:638.788	test-rmse:957.711
[4]	train-rmse:621.439	test-rmse:931.868
[5]	train-rmse:604.633	test-rmse:907.27
[6]	train-rmse:588.308	test-rmse:882.817
[7]	train-rmse:572.804	test-rmse:859.505
[8]	train-rmse:557.461	test-rmse:836.386
[9]	train-rmse:542.599	test-rmse:814.368
[10]	train-rmse:528.146	test-rmse:792.681
[11]	train-rmse:514.218	test-rmse:771.902
[12]	train-rmse:500.766	test-rmse:751.8
[13]	train-rmse:487.696	test-rmse:732.445
[14]	train-rmse:475.031	test-rmse:713.603
[15]	train-rmse:462.721	test-rmse:694.976
[16]	train-rmse:450.858	test-rmse:677.147
[17]	train-rmse:439.592	test-rmse:660.234
[18]	train-rmse:428.481	test-rmse:643.654
[19]	train-rmse:417.678	test-rmse:627.356
[20]	train-rmse:407.422	test-rm

[191]	train-rmse:123.782	test-rmse:168.39
[192]	train-rmse:123.721	test-rmse:168.256
[193]	train-rmse:123.638	test-rmse:168.112
[194]	train-rmse:123.502	test-rmse:167.942
[195]	train-rmse:123.421	test-rmse:167.81
[196]	train-rmse:123.385	test-rmse:167.66
[197]	train-rmse:123.324	test-rmse:167.588
[198]	train-rmse:123.226	test-rmse:167.348
[199]	train-rmse:123.157	test-rmse:167.237
[200]	train-rmse:123.063	test-rmse:167.019
[201]	train-rmse:123.006	test-rmse:166.901
[202]	train-rmse:122.888	test-rmse:166.775
[203]	train-rmse:122.791	test-rmse:166.672
[204]	train-rmse:122.747	test-rmse:166.576
[205]	train-rmse:122.672	test-rmse:166.499
[206]	train-rmse:122.617	test-rmse:166.367
[207]	train-rmse:122.549	test-rmse:166.256
[208]	train-rmse:122.481	test-rmse:166.097
[209]	train-rmse:122.445	test-rmse:166.004
[210]	train-rmse:122.331	test-rmse:165.847
[211]	train-rmse:122.232	test-rmse:165.74
[212]	train-rmse:122.206	test-rmse:165.676
[213]	train-rmse:122.173	test-rmse:165.578
[214]	train-rms

[383]	train-rmse:113.268	test-rmse:151.66
[384]	train-rmse:113.224	test-rmse:151.627
[385]	train-rmse:113.206	test-rmse:151.573
[386]	train-rmse:113.168	test-rmse:151.497
[387]	train-rmse:113.128	test-rmse:151.409
[388]	train-rmse:113.096	test-rmse:151.342
[389]	train-rmse:113.058	test-rmse:151.179
[390]	train-rmse:113.019	test-rmse:151.15
[391]	train-rmse:112.98	test-rmse:151.1
[392]	train-rmse:112.924	test-rmse:151.071
[393]	train-rmse:112.87	test-rmse:150.917
[394]	train-rmse:112.839	test-rmse:150.846
[395]	train-rmse:112.818	test-rmse:150.795
[396]	train-rmse:112.798	test-rmse:150.733
[397]	train-rmse:112.753	test-rmse:150.679
[398]	train-rmse:112.736	test-rmse:150.644
[399]	train-rmse:112.722	test-rmse:150.593
[400]	train-rmse:112.698	test-rmse:150.548
[401]	train-rmse:112.68	test-rmse:150.493
[402]	train-rmse:112.657	test-rmse:150.461
[403]	train-rmse:112.629	test-rmse:150.42
[404]	train-rmse:112.587	test-rmse:150.348
[405]	train-rmse:112.539	test-rmse:150.269
[406]	train-rmse:11

[575]	train-rmse:107.463	test-rmse:142.598
[576]	train-rmse:107.421	test-rmse:142.543
[577]	train-rmse:107.405	test-rmse:142.501
[578]	train-rmse:107.374	test-rmse:142.473
[579]	train-rmse:107.336	test-rmse:142.415
[580]	train-rmse:107.312	test-rmse:142.38
[581]	train-rmse:107.292	test-rmse:142.345
[582]	train-rmse:107.258	test-rmse:142.308
[583]	train-rmse:107.231	test-rmse:142.266
[584]	train-rmse:107.202	test-rmse:142.245
[585]	train-rmse:107.194	test-rmse:142.212
[586]	train-rmse:107.165	test-rmse:142.18
[587]	train-rmse:107.134	test-rmse:142.149
[588]	train-rmse:107.108	test-rmse:142.107
[589]	train-rmse:107.1	test-rmse:142.08
[590]	train-rmse:107.081	test-rmse:142.05
[591]	train-rmse:107.058	test-rmse:142.008
[592]	train-rmse:107.013	test-rmse:141.956
[593]	train-rmse:106.978	test-rmse:141.911
[594]	train-rmse:106.947	test-rmse:141.871
[595]	train-rmse:106.916	test-rmse:141.839
[596]	train-rmse:106.881	test-rmse:141.796
[597]	train-rmse:106.857	test-rmse:141.769
[598]	train-rmse:

[767]	train-rmse:103.129	test-rmse:136.069
[768]	train-rmse:103.122	test-rmse:136.043
[769]	train-rmse:103.103	test-rmse:136.019
[770]	train-rmse:103.084	test-rmse:135.982
[771]	train-rmse:103.075	test-rmse:135.951
[772]	train-rmse:103.053	test-rmse:135.933
[773]	train-rmse:103.027	test-rmse:135.896
[774]	train-rmse:103.006	test-rmse:135.873
[775]	train-rmse:103.001	test-rmse:135.851
[776]	train-rmse:102.979	test-rmse:135.816
[777]	train-rmse:102.97	test-rmse:135.787
[778]	train-rmse:102.963	test-rmse:135.767
[779]	train-rmse:102.931	test-rmse:135.727
[780]	train-rmse:102.905	test-rmse:135.686
[781]	train-rmse:102.873	test-rmse:135.64
[782]	train-rmse:102.846	test-rmse:135.618
[783]	train-rmse:102.83	test-rmse:135.597
[784]	train-rmse:102.811	test-rmse:135.562
[785]	train-rmse:102.8	test-rmse:135.545
[786]	train-rmse:102.782	test-rmse:135.51
[787]	train-rmse:102.751	test-rmse:135.495
[788]	train-rmse:102.741	test-rmse:135.482
[789]	train-rmse:102.724	test-rmse:135.469
[790]	train-rmse:

[959]	train-rmse:99.7617	test-rmse:131.194
[960]	train-rmse:99.7493	test-rmse:131.174
[961]	train-rmse:99.7409	test-rmse:131.156
[962]	train-rmse:99.7265	test-rmse:131.128
[963]	train-rmse:99.7088	test-rmse:131.114
[964]	train-rmse:99.6885	test-rmse:131.092
[965]	train-rmse:99.674	test-rmse:131.088
[966]	train-rmse:99.654	test-rmse:131.075
[967]	train-rmse:99.6415	test-rmse:131.059
[968]	train-rmse:99.6219	test-rmse:131.028
[969]	train-rmse:99.6004	test-rmse:131.008
[970]	train-rmse:99.5838	test-rmse:130.98
[971]	train-rmse:99.5695	test-rmse:130.968
[972]	train-rmse:99.5349	test-rmse:130.935
[973]	train-rmse:99.5224	test-rmse:130.912
[974]	train-rmse:99.5017	test-rmse:130.887
[975]	train-rmse:99.4769	test-rmse:130.858
[976]	train-rmse:99.4638	test-rmse:130.85
[977]	train-rmse:99.4446	test-rmse:130.836
[978]	train-rmse:99.4163	test-rmse:130.803
[979]	train-rmse:99.4003	test-rmse:130.798
[980]	train-rmse:99.3897	test-rmse:130.777
[981]	train-rmse:99.3806	test-rmse:130.753
[982]	train-rms

## 3. 11月份预测导出

In [42]:
nov_data = pd.read_csv('../datasets/Nov_test_data.csv')

In [43]:
%time nov_data = real_time_predict(model_xgb,nov_data)

D:\Anaconda\lib\site-packages\ipykernel_launcher.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  


[ 419.]
第0行预测完成！
[ 359.]
第1行预测完成！
[ 327.]
第2行预测完成！
[ 296.]
第3行预测完成！
[ 303.]
第4行预测完成！
[ 313.]
第5行预测完成！
[ 290.]
第6行预测完成！
[ 513.]
第7行预测完成！
[ 693.]
第8行预测完成！
[ 654.]
第9行预测完成！
[ 535.]
第10行预测完成！
[ 755.]
第11行预测完成！
[ 1068.]
第12行预测完成！
[ 829.]
第13行预测完成！
[ 598.]
第14行预测完成！
[ 547.]
第15行预测完成！
[ 548.]
第16行预测完成！
[ 899.]
第17行预测完成！
[ 829.]
第18行预测完成！
[ 599.]
第19行预测完成！
[ 558.]
第20行预测完成！
[ 610.]
第21行预测完成！
[ 639.]
第22行预测完成！
[ 415.]
第23行预测完成！
[ 360.]
第24行预测完成！
[ 317.]
第25行预测完成！
[ 274.]
第26行预测完成！
[ 274.]
第27行预测完成！
[ 262.]
第28行预测完成！
[ 293.]
第29行预测完成！
[ 293.]
第30行预测完成！
[ 536.]
第31行预测完成！
[ 684.]
第32行预测完成！
[ 630.]
第33行预测完成！
[ 550.]
第34行预测完成！
[ 810.]
第35行预测完成！
[ 1053.]
第36行预测完成！
[ 835.]
第37行预测完成！
[ 610.]
第38行预测完成！
[ 565.]
第39行预测完成！
[ 539.]
第40行预测完成！
[ 892.]
第41行预测完成！
[ 842.]
第42行预测完成！
[ 635.]
第43行预测完成！
[ 513.]
第44行预测完成！
[ 574.]
第45行预测完成！
[ 585.]
第46行预测完成！
[ 387.]
第47行预测完成！
[ 292.]
第48行预测完成！
[ 338.]
第49行预测完成！
[ 288.]
第50行预测完成！
[ 283.]
第51行预测完成！
[ 273.]
第52行预测完成！
[ 279.]
第53行预测完成！
[ 286.]
第54行预测完成！
[ 468.]
第55行预测完成！


第441行预测完成！
[ 388.]
第442行预测完成！
[ 653.]
第443行预测完成！
[ 958.]
第444行预测完成！
[ 665.]
第445行预测完成！
[ 509.]
第446行预测完成！
[ 425.]
第447行预测完成！
[ 405.]
第448行预测完成！
[ 732.]
第449行预测完成！
[ 755.]
第450行预测完成！
[ 552.]
第451行预测完成！
[ 407.]
第452行预测完成！
[ 394.]
第453行预测完成！
[ 390.]
第454行预测完成！
[ 262.]
第455行预测完成！
[ 221.]
第456行预测完成！
[ 201.]
第457行预测完成！
[ 202.]
第458行预测完成！
[ 224.]
第459行预测完成！
[ 223.]
第460行预测完成！
[ 247.]
第461行预测完成！
[ 300.]
第462行预测完成！
[ 493.]
第463行预测完成！
[ 623.]
第464行预测完成！
[ 560.]
第465行预测完成！
[ 463.]
第466行预测完成！
[ 779.]
第467行预测完成！
[ 1117.]
第468行预测完成！
[ 840.]
第469行预测完成！
[ 615.]
第470行预测完成！
[ 521.]
第471行预测完成！
[ 484.]
第472行预测完成！
[ 897.]
第473行预测完成！
[ 894.]
第474行预测完成！
[ 632.]
第475行预测完成！
[ 511.]
第476行预测完成！
[ 513.]
第477行预测完成！
[ 536.]
第478行预测完成！
[ 348.]
第479行预测完成！
[ 262.]
第480行预测完成！
[ 222.]
第481行预测完成！
[ 211.]
第482行预测完成！
[ 221.]
第483行预测完成！
[ 227.]
第484行预测完成！
[ 246.]
第485行预测完成！
[ 290.]
第486行预测完成！
[ 565.]
第487行预测完成！
[ 710.]
第488行预测完成！
[ 619.]
第489行预测完成！
[ 498.]
第490行预测完成！
[ 824.]
第491行预测完成！
[ 1172.]
第492行预测完成！
[ 878.]
第493行预测完成！

第158行预测完成！
[ 643.]
第159行预测完成！
[ 510.]
第160行预测完成！
[ 955.]
第161行预测完成！
[ 876.]
第162行预测完成！
[ 609.]
第163行预测完成！
[ 530.]
第164行预测完成！
[ 554.]
第165行预测完成！
[ 673.]
第166行预测完成！
[ 574.]
第167行预测完成！
[ 591.]
第168行预测完成！
[ 685.]
第169行预测完成！
[ 743.]
第170行预测完成！
[ 835.]
第171行预测完成！
[ 951.]
第172行预测完成！
[ 1025.]
第173行预测完成！
[ 935.]
第174行预测完成！
[ 895.]
第175行预测完成！
[ 750.]
第176行预测完成！
[ 586.]
第177行预测完成！
[ 458.]
第178行预测完成！
[ 740.]
第179行预测完成！
[ 1121.]
第180行预测完成！
[ 913.]
第181行预测完成！
[ 647.]
第182行预测完成！
[ 595.]
第183行预测完成！
[ 489.]
第184行预测完成！
[ 908.]
第185行预测完成！
[ 859.]
第186行预测完成！
[ 598.]
第187行预测完成！
[ 496.]
第188行预测完成！
[ 536.]
第189行预测完成！
[ 638.]
第190行预测完成！
[ 536.]
第191行预测完成！
[ 530.]
第192行预测完成！
[ 670.]
第193行预测完成！
[ 727.]
第194行预测完成！
[ 834.]
第195行预测完成！
[ 970.]
第196行预测完成！
[ 996.]
第197行预测完成！
[ 922.]
第198行预测完成！
[ 864.]
第199行预测完成！
[ 732.]
第200行预测完成！
[ 583.]
第201行预测完成！
[ 469.]
第202行预测完成！
[ 732.]
第203行预测完成！
[ 1095.]
第204行预测完成！
[ 908.]
第205行预测完成！
[ 662.]
第206行预测完成！
[ 597.]
第207行预测完成！
[ 492.]
第208行预测完成！
[ 936.]
第209行预测完成！
[ 876.]
第210行预测完成

第592行预测完成！
[ 710.]
第593行预测完成！
[ 720.]
第594行预测完成！
[ 510.]
第595行预测完成！
[ 410.]
第596行预测完成！
[ 406.]
第597行预测完成！
[ 479.]
第598行预测完成！
[ 386.]
第599行预测完成！
[ 389.]
第600行预测完成！
[ 442.]
第601行预测完成！
[ 554.]
第602行预测完成！
[ 665.]
第603行预测完成！
[ 761.]
第604行预测完成！
[ 938.]
第605行预测完成！
[ 988.]
第606行预测完成！
[ 982.]
第607行预测完成！
[ 932.]
第608行预测完成！
[ 772.]
第609行预测完成！
[ 547.]
第610行预测完成！
[ 812.]
第611行预测完成！
[ 1057.]
第612行预测完成！
[ 896.]
第613行预测完成！
[ 678.]
第614行预测完成！
[ 555.]
第615行预测完成！
[ 454.]
第616行预测完成！
[ 738.]
第617行预测完成！
[ 749.]
第618行预测完成！
[ 536.]
第619行预测完成！
[ 404.]
第620行预测完成！
[ 391.]
第621行预测完成！
[ 448.]
第622行预测完成！
[ 390.]
第623行预测完成！
[ 454.]
第624行预测完成！
[ 547.]
第625行预测完成！
[ 688.]
第626行预测完成！
[ 786.]
第627行预测完成！
[ 897.]
第628行预测完成！
[ 1090.]
第629行预测完成！
[ 1160.]
第630行预测完成！
[ 1198.]
第631行预测完成！
[ 1157.]
第632行预测完成！
[ 918.]
第633行预测完成！
[ 663.]
第634行预测完成！
[ 939.]
第635行预测完成！
[ 1207.]
第636行预测完成！
[ 1035.]
第637行预测完成！
[ 800.]
第638行预测完成！
[ 644.]
第639行预测完成！
[ 536.]
第640行预测完成！
[ 909.]
第641行预测完成！
[ 876.]
第642行预测完成！
[ 610.]
第643行预测完成！
[ 496.]
第644行

第301行预测完成！
[ 1186.]
第302行预测完成！
[ 1128.]
第303行预测完成！
[ 1120.]
第304行预测完成！
[ 2324.]
第305行预测完成！
[ 2172.]
第306行预测完成！
[ 1429.]
第307行预测完成！
[ 1191.]
第308行预测完成！
[ 1106.]
第309行预测完成！
[ 1163.]
第310行预测完成！
[ 515.]
第311行预测完成！
[ 118.]
第312行预测完成！
[ 51.]
第313行预测完成！
[ 23.]
第314行预测完成！
[ 13.]
第315行预测完成！
[ 11.]
第316行预测完成！
[ 23.]
第317行预测完成！
[ 141.]
第318行预测完成！
[ 926.]
第319行预测完成！
[ 1498.]
第320行预测完成！
[ 1198.]
第321行预测完成！
[ 990.]
第322行预测完成！
[ 2041.]
第323行预测完成！
[ 2742.]
第324行预测完成！
[ 1700.]
第325行预测完成！
[ 1263.]
第326行预测完成！
[ 1132.]
第327行预测完成！
[ 1140.]
第328行预测完成！
[ 2243.]
第329行预测完成！
[ 2092.]
第330行预测完成！
[ 1447.]
第331行预测完成！
[ 1202.]
第332行预测完成！
[ 1119.]
第333行预测完成！
[ 1185.]
第334行预测完成！
[ 509.]
第335行预测完成！
[ 132.]
第336行预测完成！
[ 66.]
第337行预测完成！
[ 29.]
第338行预测完成！
[ 16.]
第339行预测完成！
[ 13.]
第340行预测完成！
[ 25.]
第341行预测完成！
[ 133.]
第342行预测完成！
[ 930.]
第343行预测完成！
[ 1480.]
第344行预测完成！
[ 1188.]
第345行预测完成！
[ 980.]
第346行预测完成！
[ 2027.]
第347行预测完成！
[ 2831.]
第348行预测完成！
[ 1751.]
第349行预测完成！
[ 1252.]
第350行预测完成！
[ 1136.]
第351行预测完成！
[ 1175.]
第352行预测完成！

第8行预测完成！
[ 506.]
第9行预测完成！
[ 379.]
第10行预测完成！
[ 780.]
第11行预测完成！
[ 1159.]
第12行预测完成！
[ 775.]
第13行预测完成！
[ 569.]
第14行预测完成！
[ 456.]
第15行预测完成！
[ 402.]
第16行预测完成！
[ 865.]
第17行预测完成！
[ 909.]
第18行预测完成！
[ 612.]
第19行预测完成！
[ 536.]
第20行预测完成！
[ 565.]
第21行预测完成！
[ 689.]
第22行预测完成！
[ 480.]
第23行预测完成！
[ 304.]
第24行预测完成！
[ 213.]
第25行预测完成！
[ 173.]
第26行预测完成！
[ 140.]
第27行预测完成！
[ 128.]
第28行预测完成！
[ 100.]
第29行预测完成！
[ 127.]
第30行预测完成！
[ 278.]
第31行预测完成！
[ 579.]
第32行预测完成！
[ 513.]
第33行预测完成！
[ 390.]
第34行预测完成！
[ 761.]
第35行预测完成！
[ 1188.]
第36行预测完成！
[ 771.]
第37行预测完成！
[ 577.]
第38行预测完成！
[ 473.]
第39行预测完成！
[ 402.]
第40行预测完成！
[ 872.]
第41行预测完成！
[ 890.]
第42行预测完成！
[ 619.]
第43行预测完成！
[ 521.]
第44行预测完成！
[ 591.]
第45行预测完成！
[ 666.]
第46行预测完成！
[ 439.]
第47行预测完成！
[ 267.]
第48行预测完成！
[ 210.]
第49行预测完成！
[ 181.]
第50行预测完成！
[ 170.]
第51行预测完成！
[ 165.]
第52行预测完成！
[ 153.]
第53行预测完成！
[ 188.]
第54行预测完成！
[ 325.]
第55行预测完成！
[ 517.]
第56行预测完成！
[ 501.]
第57行预测完成！
[ 414.]
第58行预测完成！
[ 748.]
第59行预测完成！
[ 1165.]
第60行预测完成！
[ 779.]
第61行预测完成！
[ 581.]
第62行预测完成！
[ 502.]
第63行预测完成！

第449行预测完成！
[ 766.]
第450行预测完成！
[ 541.]
第451行预测完成！
[ 402.]
第452行预测完成！
[ 400.]
第453行预测完成！
[ 462.]
第454行预测完成！
[ 314.]
第455行预测完成！
[ 195.]
第456行预测完成！
[ 159.]
第457行预测完成！
[ 141.]
第458行预测完成！
[ 129.]
第459行预测完成！
[ 122.]
第460行预测完成！
[ 116.]
第461行预测完成！
[ 140.]
第462行预测完成！
[ 249.]
第463行预测完成！
[ 476.]
第464行预测完成！
[ 430.]
第465行预测完成！
[ 359.]
第466行预测完成！
[ 741.]
第467行预测完成！
[ 1094.]
第468行预测完成！
[ 771.]
第469行预测完成！
[ 561.]
第470行预测完成！
[ 467.]
第471行预测完成！
[ 406.]
第472行预测完成！
[ 878.]
第473行预测完成！
[ 894.]
第474行预测完成！
[ 620.]
第475行预测完成！
[ 494.]
第476行预测完成！
[ 514.]
第477行预测完成！
[ 618.]
第478行预测完成！
[ 414.]
第479行预测完成！
[ 243.]
第480行预测完成！
[ 175.]
第481行预测完成！
[ 152.]
第482行预测完成！
[ 137.]
第483行预测完成！
[ 128.]
第484行预测完成！
[ 122.]
第485行预测完成！
[ 149.]
第486行预测完成！
[ 313.]
第487行预测完成！
[ 560.]
第488行预测完成！
[ 480.]
第489行预测完成！
[ 358.]
第490行预测完成！
[ 750.]
第491行预测完成！
[ 1152.]
第492行预测完成！
[ 810.]
第493行预测完成！
[ 591.]
第494行预测完成！
[ 474.]
第495行预测完成！
[ 398.]
第496行预测完成！
[ 880.]
第497行预测完成！
[ 924.]
第498行预测完成！
[ 618.]
第499行预测完成！
[ 492.]
第500行预测完成！
[ 512.]
第501行预测完成！

第164行预测完成！
[ 614.]
第165行预测完成！
[ 638.]
第166行预测完成！
[ 368.]
第167行预测完成！
[ 220.]
第168行预测完成！
[ 157.]
第169行预测完成！
[ 138.]
第170行预测完成！
[ 121.]
第171行预测完成！
[ 108.]
第172行预测完成！
[ 109.]
第173行预测完成！
[ 158.]
第174行预测完成！
[ 388.]
第175行预测完成！
[ 564.]
第176行预测完成！
[ 541.]
第177行预测完成！
[ 457.]
第178行预测完成！
[ 732.]
第179行预测完成！
[ 1037.]
第180行预测完成！
[ 730.]
第181行预测完成！
[ 532.]
第182行预测完成！
[ 449.]
第183行预测完成！
[ 479.]
第184行预测完成！
[ 884.]
第185行预测完成！
[ 825.]
第186行预测完成！
[ 591.]
第187行预测完成！
[ 503.]
第188行预测完成！
[ 574.]
第189行预测完成！
[ 621.]
第190行预测完成！
[ 378.]
第191行预测完成！
[ 209.]
第192行预测完成！
[ 151.]
第193行预测完成！
[ 133.]
第194行预测完成！
[ 118.]
第195行预测完成！
[ 102.]
第196行预测完成！
[ 104.]
第197行预测完成！
[ 150.]
第198行预测完成！
[ 374.]
第199行预测完成！
[ 589.]
第200行预测完成！
[ 540.]
第201行预测完成！
[ 448.]
第202行预测完成！
[ 743.]
第203行预测完成！
[ 1022.]
第204行预测完成！
[ 746.]
第205行预测完成！
[ 531.]
第206行预测完成！
[ 489.]
第207行预测完成！
[ 470.]
第208行预测完成！
[ 890.]
第209行预测完成！
[ 812.]
第210行预测完成！
[ 588.]
第211行预测完成！
[ 491.]
第212行预测完成！
[ 551.]
第213行预测完成！
[ 594.]
第214行预测完成！
[ 358.]
第215行预测完成！
[ 204.]
第216行预测完成！

第598行预测完成！
[ 286.]
第599行预测完成！
[ 153.]
第600行预测完成！
[ 115.]
第601行预测完成！
[ 87.]
第602行预测完成！
[ 69.]
第603行预测完成！
[ 57.]
第604行预测完成！
[ 56.]
第605行预测完成！
[ 88.]
第606行预测完成！
[ 158.]
第607行预测完成！
[ 299.]
第608行预测完成！
[ 379.]
第609行预测完成！
[ 338.]
第610行预测完成！
[ 620.]
第611行预测完成！
[ 950.]
第612行预测完成！
[ 654.]
第613行预测完成！
[ 454.]
第614行预测完成！
[ 378.]
第615行预测完成！
[ 361.]
第616行预测完成！
[ 718.]
第617行预测完成！
[ 724.]
第618行预测完成！
[ 522.]
第619行预测完成！
[ 378.]
第620行预测完成！
[ 374.]
第621行预测完成！
[ 397.]
第622行预测完成！
[ 259.]
第623行预测完成！
[ 149.]
第624行预测完成！
[ 116.]
第625行预测完成！
[ 95.]
第626行预测完成！
[ 79.]
第627行预测完成！
[ 70.]
第628行预测完成！
[ 67.]
第629行预测完成！
[ 94.]
第630行预测完成！
[ 203.]
第631行预测完成！
[ 480.]
第632行预测完成！
[ 472.]
第633行预测完成！
[ 375.]
第634行预测完成！
[ 751.]
第635行预测完成！
[ 1093.]
第636行预测完成！
[ 762.]
第637行预测完成！
[ 552.]
第638行预测完成！
[ 460.]
第639行预测完成！
[ 415.]
第640行预测完成！
[ 873.]
第641行预测完成！
[ 855.]
第642行预测完成！
[ 581.]
第643行预测完成！
[ 475.]
第644行预测完成！
[ 496.]
第645行预测完成！
[ 546.]
第646行预测完成！
[ 346.]
第647行预测完成！
[ 183.]
第648行预测完成！
[ 129.]
第649行预测完成！
[ 99.]
第650行预测完成！
[ 89.]
第651

第315行预测完成！
[ 161.]
第316行预测完成！
[ 163.]
第317行预测完成！
[ 204.]
第318行预测完成！
[ 552.]
第319行预测完成！
[ 702.]
第320行预测完成！
[ 643.]
第321行预测完成！
[ 487.]
第322行预测完成！
[ 932.]
第323行预测完成！
[ 1455.]
第324行预测完成！
[ 961.]
第325行预测完成！
[ 653.]
第326行预测完成！
[ 549.]
第327行预测完成！
[ 505.]
第328行预测完成！
[ 1078.]
第329行预测完成！
[ 1213.]
第330行预测完成！
[ 857.]
第331行预测完成！
[ 683.]
第332行预测完成！
[ 704.]
第333行预测完成！
[ 784.]
第334行预测完成！
[ 513.]
第335行预测完成！
[ 339.]
第336行预测完成！
[ 252.]
第337行预测完成！
[ 206.]
第338行预测完成！
[ 173.]
第339行预测完成！
[ 158.]
第340行预测完成！
[ 164.]
第341行预测完成！
[ 209.]
第342行预测完成！
[ 560.]
第343行预测完成！
[ 691.]
第344行预测完成！
[ 613.]
第345行预测完成！
[ 453.]
第346行预测完成！
[ 904.]
第347行预测完成！
[ 1368.]
第348行预测完成！
[ 955.]
第349行预测完成！
[ 653.]
第350行预测完成！
[ 539.]
第351行预测完成！
[ 495.]
第352行预测完成！
[ 1032.]
第353行预测完成！
[ 1204.]
第354行预测完成！
[ 873.]
第355行预测完成！
[ 685.]
第356行预测完成！
[ 686.]
第357行预测完成！
[ 743.]
第358行预测完成！
[ 488.]
第359行预测完成！
[ 326.]
第360行预测完成！
[ 243.]
第361行预测完成！
[ 198.]
第362行预测完成！
[ 165.]
第363行预测完成！
[ 152.]
第364行预测完成！
[ 160.]
第365行预测完成！
[ 206.]
第366行预测完成！
[ 534.]
第367行预

第23行预测完成！
[ 123.]
第24行预测完成！
[ 71.]
第25行预测完成！
[ 41.]
第26行预测完成！
[ 32.]
第27行预测完成！
[ 27.]
第28行预测完成！
[ 30.]
第29行预测完成！
[ 58.]
第30行预测完成！
[ 169.]
第31行预测完成！
[ 322.]
第32行预测完成！
[ 281.]
第33行预测完成！
[ 223.]
第34行预测完成！
[ 352.]
第35行预测完成！
[ 549.]
第36行预测完成！
[ 378.]
第37行预测完成！
[ 265.]
第38行预测完成！
[ 264.]
第39行预测完成！
[ 234.]
第40行预测完成！
[ 398.]
第41行预测完成！
[ 387.]
第42行预测完成！
[ 299.]
第43行预测完成！
[ 293.]
第44行预测完成！
[ 286.]
第45行预测完成！
[ 234.]
第46行预测完成！
[ 148.]
第47行预测完成！
[ 100.]
第48行预测完成！
[ 64.]
第49行预测完成！
[ 32.]
第50行预测完成！
[ 21.]
第51行预测完成！
[ 17.]
第52行预测完成！
[ 21.]
第53行预测完成！
[ 57.]
第54行预测完成！
[ 189.]
第55行预测完成！
[ 311.]
第56行预测完成！
[ 267.]
第57行预测完成！
[ 213.]
第58行预测完成！
[ 338.]
第59行预测完成！
[ 518.]
第60行预测完成！
[ 368.]
第61行预测完成！
[ 271.]
第62行预测完成！
[ 247.]
第63行预测完成！
[ 244.]
第64行预测完成！
[ 393.]
第65行预测完成！
[ 384.]
第66行预测完成！
[ 303.]
第67行预测完成！
[ 297.]
第68行预测完成！
[ 274.]
第69行预测完成！
[ 215.]
第70行预测完成！
[ 118.]
第71行预测完成！
[ 75.]
第72行预测完成！
[ 46.]
第73行预测完成！
[ 19.]
第74行预测完成！
[ 5.]
第75行预测完成！
[ 0.]
第76行预测完成！
[ 8.]
第77行预测完成！
[ 35.]
第78行预测完成！
[ 97.]
第79行预测完成！
[ 201

第474行预测完成！
[ 220.]
第475行预测完成！
[ 185.]
第476行预测完成！
[ 166.]
第477行预测完成！
[ 135.]
第478行预测完成！
[ 66.]
第479行预测完成！
[ 24.]
第480行预测完成！
[ 13.]
第481行预测完成！
[ 10.]
第482行预测完成！
[ 7.]
第483行预测完成！
[ 7.]
第484行预测完成！
[ 12.]
第485行预测完成！
[ 32.]
第486行预测完成！
[ 120.]
第487行预测完成！
[ 250.]
第488行预测完成！
[ 255.]
第489行预测完成！
[ 192.]
第490行预测完成！
[ 288.]
第491行预测完成！
[ 408.]
第492行预测完成！
[ 317.]
第493行预测完成！
[ 241.]
第494行预测完成！
[ 207.]
第495行预测完成！
[ 199.]
第496行预测完成！
[ 357.]
第497行预测完成！
[ 326.]
第498行预测完成！
[ 223.]
第499行预测完成！
[ 191.]
第500行预测完成！
[ 173.]
第501行预测完成！
[ 137.]
第502行预测完成！
[ 63.]
第503行预测完成！
[ 19.]
第504行预测完成！
[ 6.]
第505行预测完成！
[ 4.]
第506行预测完成！
[ 3.]
第507行预测完成！
[ 3.]
第508行预测完成！
[ 8.]
第509行预测完成！
[ 28.]
第510行预测完成！
[ 109.]
第511行预测完成！
[ 250.]
第512行预测完成！
[ 252.]
第513行预测完成！
[ 201.]
第514行预测完成！
[ 292.]
第515行预测完成！
[ 405.]
第516行预测完成！
[ 315.]
第517行预测完成！
[ 236.]
第518行预测完成！
[ 208.]
第519行预测完成！
[ 207.]
第520行预测完成！
[ 359.]
第521行预测完成！
[ 322.]
第522行预测完成！
[ 219.]
第523行预测完成！
[ 186.]
第524行预测完成！
[ 161.]
第525行预测完成！
[ 132.]
第526行预测完成！
[ 53.]
第527行预测完成！
[ 14.]

[ 346.]
第198行预测完成！
[ 2634.]
第199行预测完成！
[ 3822.]
第200行预测完成！
[ 2708.]
第201行预测完成！
[ 2057.]
第202行预测完成！
[ 6164.]
第203行预测完成！
[ 7970.]
第204行预测完成！
[ 3802.]
第205行预测完成！
[ 2157.]
第206行预测完成！
[ 2322.]
第207行预测完成！
[ 2208.]
第208行预测完成！
[ 6217.]
第209行预测完成！
[ 5499.]
第210行预测完成！
[ 3292.]
第211行预测完成！
[ 2771.]
第212行预测完成！
[ 2236.]
第213行预测完成！
[ 2079.]
第214行预测完成！
[ 674.]
第215行预测完成！
[ 125.]
第216行预测完成！
[ 71.]
第217行预测完成！
[ 50.]
第218行预测完成！
[ 44.]
第219行预测完成！
[ 49.]
第220行预测完成！
[ 76.]
第221行预测完成！
[ 277.]
第222行预测完成！
[ 2174.]
第223行预测完成！
[ 3547.]
第224行预测完成！
[ 2765.]
第225行预测完成！
[ 2099.]
第226行预测完成！
[ 5838.]
第227行预测完成！
[ 7284.]
第228行预测完成！
[ 3565.]
第229行预测完成！
[ 2115.]
第230行预测完成！
[ 2248.]
第231行预测完成！
[ 2139.]
第232行预测完成！
[ 5518.]
第233行预测完成！
[ 5318.]
第234行预测完成！
[ 3140.]
第235行预测完成！
[ 2435.]
第236行预测完成！
[ 2104.]
第237行预测完成！
[ 2018.]
第238行预测完成！
[ 628.]
第239行预测完成！
[ 101.]
第240行预测完成！
[ 32.]
第241行预测完成！
[ 21.]
第242行预测完成！
[ 11.]
第243行预测完成！
[ 17.]
第244行预测完成！
[ 45.]
第245行预测完成！
[ 244.]
第246行预测完成！
[ 1238.]
第247行预测完成！
[ 2434.]
第248行预测完成！
[ 2446.

第621行预测完成！
[ 1520.]
第622行预测完成！
[ 479.]
第623行预测完成！
[ 48.]
第624行预测完成！
[ 17.]
第625行预测完成！
[ 13.]
第626行预测完成！
[ 11.]
第627行预测完成！
[ 8.]
第628行预测完成！
[ 28.]
第629行预测完成！
[ 192.]
第630行预测完成！
[ 1865.]
第631行预测完成！
[ 3162.]
第632行预测完成！
[ 2744.]
第633行预测完成！
[ 2040.]
第634行预测完成！
[ 6097.]
第635行预测完成！
[ 7743.]
第636行预测完成！
[ 3861.]
第637行预测完成！
[ 2187.]
第638行预测完成！
[ 2140.]
第639行预测完成！
[ 2205.]
第640行预测完成！
[ 6808.]
第641行预测完成！
[ 5386.]
第642行预测完成！
[ 3321.]
第643行预测完成！
[ 2537.]
第644行预测完成！
[ 1883.]
第645行预测完成！
[ 2111.]
第646行预测完成！
[ 585.]
第647行预测完成！
[ 40.]
第648行预测完成！
[-10.]
第649行预测完成！
[-10.]
第650行预测完成！
[-13.]
第651行预测完成！
[ 7.]
第652行预测完成！
[ 33.]
第653行预测完成！
[ 227.]
第654行预测完成！
[ 2228.]
第655行预测完成！
[ 3330.]
第656行预测完成！
[ 2755.]
第657行预测完成！
[ 2049.]
第658行预测完成！
[ 6159.]
第659行预测完成！
[ 7964.]
第660行预测完成！
[ 3644.]
第661行预测完成！
[ 2080.]
第662行预测完成！
[ 2149.]
第663行预测完成！
[ 2131.]
第664行预测完成！
[ 6456.]
第665行预测完成！
[ 5503.]
第666行预测完成！
[ 3143.]
第667行预测完成！
[ 2350.]
第668行预测完成！
[ 1803.]
第669行预测完成！
[ 1974.]
第670行预测完成！
[ 580.]
第671行预测完成！
[ 63.]
第672行预测完成！
[ 

第333行预测完成！
[ 169.]
第334行预测完成！
[ 171.]
第335行预测完成！
[ 185.]
第336行预测完成！
[ 178.]
第337行预测完成！
[ 198.]
第338行预测完成！
[ 170.]
第339行预测完成！
[ 154.]
第340行预测完成！
[ 145.]
第341行预测完成！
[ 145.]
第342行预测完成！
[ 191.]
第343行预测完成！
[ 227.]
第344行预测完成！
[ 192.]
第345行预测完成！
[ 154.]
第346行预测完成！
[ 237.]
第347行预测完成！
[ 330.]
第348行预测完成！
[ 260.]
第349行预测完成！
[ 206.]
第350行预测完成！
[ 178.]
第351行预测完成！
[ 158.]
第352行预测完成！
[ 236.]
第353行预测完成！
[ 232.]
第354行预测完成！
[ 183.]
第355行预测完成！
[ 179.]
第356行预测完成！
[ 180.]
第357行预测完成！
[ 173.]
第358行预测完成！
[ 158.]
第359行预测完成！
[ 180.]
第360行预测完成！
[ 176.]
第361行预测完成！
[ 180.]
第362行预测完成！
[ 167.]
第363行预测完成！
[ 160.]
第364行预测完成！
[ 154.]
第365行预测完成！
[ 153.]
第366行预测完成！
[ 204.]
第367行预测完成！
[ 230.]
第368行预测完成！
[ 189.]
第369行预测完成！
[ 149.]
第370行预测完成！
[ 228.]
第371行预测完成！
[ 327.]
第372行预测完成！
[ 258.]
第373行预测完成！
[ 203.]
第374行预测完成！
[ 166.]
第375行预测完成！
[ 155.]
第376行预测完成！
[ 224.]
第377行预测完成！
[ 218.]
第378行预测完成！
[ 169.]
第379行预测完成！
[ 155.]
第380行预测完成！
[ 162.]
第381行预测完成！
[ 155.]
第382行预测完成！
[ 133.]
第383行预测完成！
[ 137.]
第384行预测完成！
[ 145.]
第385行预测完成！
[

第53行预测完成！
[ 145.]
第54行预测完成！
[ 915.]
第55行预测完成！
[ 1579.]
第56行预测完成！
[ 1067.]
第57行预测完成！
[ 956.]
第58行预测完成！
[ 3266.]
第59行预测完成！
[ 4036.]
第60行预测完成！
[ 1517.]
第61行预测完成！
[ 728.]
第62行预测完成！
[ 769.]
第63行预测完成！
[ 923.]
第64行预测完成！
[ 2811.]
第65行预测完成！
[ 2057.]
第66行预测完成！
[ 1248.]
第67行预测完成！
[ 878.]
第68行预测完成！
[ 577.]
第69行预测完成！
[ 316.]
第70行预测完成！
[ 83.]
第71行预测完成！
[-2.]
第72行预测完成！
[-15.]
第73行预测完成！
[-10.]
第74行预测完成！
[-7.]
第75行预测完成！
[-0.]
第76行预测完成！
[ 18.]
第77行预测完成！
[ 104.]
第78行预测完成！
[ 465.]
第79行预测完成！
[ 901.]
第80行预测完成！
[ 950.]
第81行预测完成！
[ 986.]
第82行预测完成！
[ 1890.]
第83行预测完成！
[ 2255.]
第84行预测完成！
[ 1291.]
第85行预测完成！
[ 908.]
第86行预测完成！
[ 897.]
第87行预测完成！
[ 1084.]
第88行预测完成！
[ 2063.]
第89行预测完成！
[ 1746.]
第90行预测完成！
[ 1157.]
第91行预测完成！
[ 840.]
第92行预测完成！
[ 535.]
第93行预测完成！
[ 298.]
第94行预测完成！
[ 45.]
第95行预测完成！
[-18.]
第96行预测完成！
[-25.]
第97行预测完成！
[-2.]
第98行预测完成！
[-5.]
第99行预测完成！
[-0.]
第100行预测完成！
[ 17.]
第101行预测完成！
[ 98.]
第102行预测完成！
[ 422.]
第103行预测完成！
[ 820.]
第104行预测完成！
[ 934.]
第105行预测完成！
[ 903.]
第106行预测完成！
[ 1779.]
第107行预测完成！
[ 2556.]
第108行预

第495行预测完成！
[ 1058.]
第496行预测完成！
[ 2685.]
第497行预测完成！
[ 2116.]
第498行预测完成！
[ 1286.]
第499行预测完成！
[ 960.]
第500行预测完成！
[ 637.]
第501行预测完成！
[ 356.]
第502行预测完成！
[ 83.]
第503行预测完成！
[-5.]
第504行预测完成！
[-10.]
第505行预测完成！
[-4.]
第506行预测完成！
[-0.]
第507行预测完成！
[ 6.]
第508行预测完成！
[ 30.]
第509行预测完成！
[ 164.]
第510行预测完成！
[ 1107.]
第511行预测完成！
[ 1691.]
第512行预测完成！
[ 1197.]
第513行预测完成！
[ 1002.]
第514行预测完成！
[ 3182.]
第515行预测完成！
[ 4109.]
第516行预测完成！
[ 1574.]
第517行预测完成！
[ 970.]
第518行预测完成！
[ 879.]
第519行预测完成！
[ 1017.]
第520行预测完成！
[ 2715.]
第521行预测完成！
[ 2190.]
第522行预测完成！
[ 1260.]
第523行预测完成！
[ 944.]
第524行预测完成！
[ 624.]
第525行预测完成！
[ 351.]
第526行预测完成！
[ 75.]
第527行预测完成！
[ 5.]
第528行预测完成！
[ 2.]
第529行预测完成！
[ 4.]
第530行预测完成！
[ 8.]
第531行预测完成！
[ 13.]
第532行预测完成！
[ 38.]
第533行预测完成！
[ 171.]
第534行预测完成！
[ 1112.]
第535行预测完成！
[ 1637.]
第536行预测完成！
[ 1160.]
第537行预测完成！
[ 1012.]
第538行预测完成！
[ 3120.]
第539行预测完成！
[ 3673.]
第540行预测完成！
[ 1557.]
第541行预测完成！
[ 985.]
第542行预测完成！
[ 881.]
第543行预测完成！
[ 1070.]
第544行预测完成！
[ 2750.]
第545行预测完成！
[ 1997.]
第546行预测完成！
[ 1227.]
第547行预测完

[ 500.]
第213行预测完成！
[ 519.]
第214行预测完成！
[ 379.]
第215行预测完成！
[ 284.]
第216行预测完成！
[ 247.]
第217行预测完成！
[ 203.]
第218行预测完成！
[ 174.]
第219行预测完成！
[ 158.]
第220行预测完成！
[ 161.]
第221行预测完成！
[ 201.]
第222行预测完成！
[ 457.]
第223行预测完成！
[ 449.]
第224行预测完成！
[ 467.]
第225行预测完成！
[ 400.]
第226行预测完成！
[ 702.]
第227行预测完成！
[ 1000.]
第228行预测完成！
[ 761.]
第229行预测完成！
[ 527.]
第230行预测完成！
[ 486.]
第231行预测完成！
[ 433.]
第232行预测完成！
[ 817.]
第233行预测完成！
[ 756.]
第234行预测完成！
[ 522.]
第235行预测完成！
[ 430.]
第236行预测完成！
[ 448.]
第237行预测完成！
[ 491.]
第238行预测完成！
[ 342.]
第239行预测完成！
[ 265.]
第240行预测完成！
[ 229.]
第241行预测完成！
[ 187.]
第242行预测完成！
[ 172.]
第243行预测完成！
[ 151.]
第244行预测完成！
[ 157.]
第245行预测完成！
[ 190.]
第246行预测完成！
[ 267.]
第247行预测完成！
[ 350.]
第248行预测完成！
[ 361.]
第249行预测完成！
[ 344.]
第250行预测完成！
[ 578.]
第251行预测完成！
[ 885.]
第252行预测完成！
[ 619.]
第253行预测完成！
[ 430.]
第254行预测完成！
[ 386.]
第255行预测完成！
[ 384.]
第256行预测完成！
[ 647.]
第257行预测完成！
[ 670.]
第258行预测完成！
[ 468.]
第259行预测完成！
[ 378.]
第260行预测完成！
[ 400.]
第261行预测完成！
[ 446.]
第262行预测完成！
[ 323.]
第263行预测完成！
[ 276.]
第264行预测完成！
[ 241.]
第26

第645行预测完成！
[ 494.]
第646行预测完成！
[ 346.]
第647行预测完成！
[ 217.]
第648行预测完成！
[ 187.]
第649行预测完成！
[ 168.]
第650行预测完成！
[ 160.]
第651行预测完成！
[ 150.]
第652行预测完成！
[ 158.]
第653行预测完成！
[ 198.]
第654行预测完成！
[ 458.]
第655行预测完成！
[ 502.]
第656行预测完成！
[ 464.]
第657行预测完成！
[ 371.]
第658行预测完成！
[ 733.]
第659行预测完成！
[ 1089.]
第660行预测完成！
[ 800.]
第661行预测完成！
[ 590.]
第662行预测完成！
[ 502.]
第663行预测完成！
[ 440.]
第664行预测完成！
[ 906.]
第665行预测完成！
[ 874.]
第666行预测完成！
[ 580.]
第667行预测完成！
[ 453.]
第668行预测完成！
[ 475.]
第669行预测完成！
[ 511.]
第670行预测完成！
[ 354.]
第671行预测完成！
[ 216.]
第672行预测完成！
[ 186.]
第673行预测完成！
[ 163.]
第674行预测完成！
[ 157.]
第675行预测完成！
[ 148.]
第676行预测完成！
[ 155.]
第677行预测完成！
[ 201.]
第678行预测完成！
[ 488.]
第679行预测完成！
[ 526.]
第680行预测完成！
[ 460.]
第681行预测完成！
[ 362.]
第682行预测完成！
[ 711.]
第683行预测完成！
[ 1076.]
第684行预测完成！
[ 778.]
第685行预测完成！
[ 571.]
第686行预测完成！
[ 493.]
第687行预测完成！
[ 438.]
第688行预测完成！
[ 882.]
第689行预测完成！
[ 856.]
第690行预测完成！
[ 574.]
第691行预测完成！
[ 451.]
第692行预测完成！
[ 468.]
第693行预测完成！
[ 501.]
第694行预测完成！
[ 338.]
第695行预测完成！
[ 208.]
第696行预测完成！
[ 183.]
第697行预测完成！

第356行预测完成！
[ 1880.]
第357行预测完成！
[ 1885.]
第358行预测完成！
[ 532.]
第359行预测完成！
[ 23.]
第360行预测完成！
[-5.]
第361行预测完成！
[-6.]
第362行预测完成！
[ 8.]
第363行预测完成！
[ 22.]
第364行预测完成！
[ 71.]
第365行预测完成！
[ 375.]
第366行预测完成！
[ 3122.]
第367行预测完成！
[ 4001.]
第368行预测完成！
[ 3088.]
第369行预测完成！
[ 2153.]
第370行预测完成！
[ 6726.]
第371行预测完成！
[ 8576.]
第372行预测完成！
[ 3580.]
第373行预测完成！
[ 2176.]
第374行预测完成！
[ 2277.]
第375行预测完成！
[ 2208.]
第376行预测完成！
[ 6068.]
第377行预测完成！
[ 5298.]
第378行预测完成！
[ 3396.]
第379行预测完成！
[ 2271.]
第380行预测完成！
[ 1740.]
第381行预测完成！
[ 1742.]
第382行预测完成！
[ 494.]
第383行预测完成！
[ 48.]
第384行预测完成！
[ 2.]
第385行预测完成！
[-4.]
第386行预测完成！
[ 17.]
第387行预测完成！
[ 34.]
第388行预测完成！
[ 84.]
第389行预测完成！
[ 366.]
第390行预测完成！
[ 2823.]
第391行预测完成！
[ 3688.]
第392行预测完成！
[ 2973.]
第393行预测完成！
[ 2173.]
第394行预测完成！
[ 6107.]
第395行预测完成！
[ 7769.]
第396行预测完成！
[ 3503.]
第397行预测完成！
[ 2081.]
第398行预测完成！
[ 1981.]
第399行预测完成！
[ 2055.]
第400行预测完成！
[ 5662.]
第401行预测完成！
[ 5081.]
第402行预测完成！
[ 3074.]
第403行预测完成！
[ 1953.]
第404行预测完成！
[ 1744.]
第405行预测完成！
[ 1691.]
第406行预测完成！
[ 493.]
第407行预测完成！
[ 5

第68行预测完成！
[ 346.]
第69行预测完成！
[ 333.]
第70行预测完成！
[ 171.]
第71行预测完成！
[ 62.]
第72行预测完成！
[ 18.]
第73行预测完成！
[ 1.]
第74行预测完成！
[-5.]
第75行预测完成！
[-3.]
第76行预测完成！
[ 2.]
第77行预测完成！
[ 18.]
第78行预测完成！
[ 97.]
第79行预测完成！
[ 241.]
第80行预测完成！
[ 320.]
第81行预测完成！
[ 288.]
第82行预测完成！
[ 438.]
第83行预测完成！
[ 561.]
第84行预测完成！
[ 381.]
第85行预测完成！
[ 283.]
第86行预测完成！
[ 272.]
第87行预测完成！
[ 274.]
第88行预测完成！
[ 420.]
第89行预测完成！
[ 436.]
第90行预测完成！
[ 325.]
第91行预测完成！
[ 275.]
第92行预测完成！
[ 313.]
第93行预测完成！
[ 298.]
第94行预测完成！
[ 170.]
第95行预测完成！
[ 60.]
第96行预测完成！
[ 19.]
第97行预测完成！
[ 2.]
第98行预测完成！
[-5.]
第99行预测完成！
[-4.]
第100行预测完成！
[ 1.]
第101行预测完成！
[ 16.]
第102行预测完成！
[ 98.]
第103行预测完成！
[ 218.]
第104行预测完成！
[ 296.]
第105行预测完成！
[ 278.]
第106行预测完成！
[ 402.]
第107行预测完成！
[ 566.]
第108行预测完成！
[ 397.]
第109行预测完成！
[ 285.]
第110行预测完成！
[ 266.]
第111行预测完成！
[ 268.]
第112行预测完成！
[ 417.]
第113行预测完成！
[ 470.]
第114行预测完成！
[ 340.]
第115行预测完成！
[ 266.]
第116行预测完成！
[ 288.]
第117行预测完成！
[ 285.]
第118行预测完成！
[ 143.]
第119行预测完成！
[ 65.]
第120行预测完成！
[ 31.]
第121行预测完成！
[ 19.]
第122行预测完成！
[ 14.]
第123行预测完成！
[ 14

第513行预测完成！
[ 263.]
第514行预测完成！
[ 404.]
第515行预测完成！
[ 614.]
第516行预测完成！
[ 406.]
第517行预测完成！
[ 322.]
第518行预测完成！
[ 271.]
第519行预测完成！
[ 268.]
第520行预测完成！
[ 478.]
第521行预测完成！
[ 465.]
第522行预测完成！
[ 326.]
第523行预测完成！
[ 271.]
第524行预测完成！
[ 289.]
第525行预测完成！
[ 287.]
第526行预测完成！
[ 135.]
第527行预测完成！
[ 49.]
第528行预测完成！
[ 18.]
第529行预测完成！
[ 9.]
第530行预测完成！
[ 5.]
第531行预测完成！
[ 5.]
第532行预测完成！
[ 9.]
第533行预测完成！
[ 32.]
第534行预测完成！
[ 144.]
第535行预测完成！
[ 360.]
第536行预测完成！
[ 334.]
第537行预测完成！
[ 261.]
第538行预测完成！
[ 402.]
第539行预测完成！
[ 611.]
第540行预测完成！
[ 403.]
第541行预测完成！
[ 316.]
第542行预测完成！
[ 273.]
第543行预测完成！
[ 269.]
第544行预测完成！
[ 460.]
第545行预测完成！
[ 440.]
第546行预测完成！
[ 310.]
第547行预测完成！
[ 258.]
第548行预测完成！
[ 270.]
第549行预测完成！
[ 254.]
第550行预测完成！
[ 117.]
第551行预测完成！
[ 34.]
第552行预测完成！
[ 11.]
第553行预测完成！
[ 4.]
第554行预测完成！
[ 1.]
第555行预测完成！
[ 2.]
第556行预测完成！
[ 7.]
第557行预测完成！
[ 28.]
第558行预测完成！
[ 127.]
第559行预测完成！
[ 302.]
第560行预测完成！
[ 325.]
第561行预测完成！
[ 248.]
第562行预测完成！
[ 396.]
第563行预测完成！
[ 583.]
第564行预测完成！
[ 382.]
第565行预测完成！
[ 296.]
第566行预测完成！
[ 26

[ 76.]
第239行预测完成！
[ 8.]
第240行预测完成！
[-7.]
第241行预测完成！
[-5.]
第242行预测完成！
[-3.]
第243行预测完成！
[-0.]
第244行预测完成！
[ 8.]
第245行预测完成！
[ 43.]
第246行预测完成！
[ 147.]
第247行预测完成！
[ 292.]
第248行预测完成！
[ 355.]
第249行预测完成！
[ 346.]
第250行预测完成！
[ 434.]
第251行预测完成！
[ 461.]
第252行预测完成！
[ 335.]
第253行预测完成！
[ 321.]
第254行预测完成！
[ 321.]
第255行预测完成！
[ 326.]
第256行预测完成！
[ 434.]
第257行预测完成！
[ 381.]
第258行预测完成！
[ 294.]
第259行预测完成！
[ 255.]
第260行预测完成！
[ 239.]
第261行预测完成！
[ 192.]
第262行预测完成！
[ 70.]
第263行预测完成！
[ 13.]
第264行预测完成！
[-2.]
第265行预测完成！
[-5.]
第266行预测完成！
[-4.]
第267行预测完成！
[-2.]
第268行预测完成！
[ 2.]
第269行预测完成！
[ 23.]
第270行预测完成！
[ 107.]
第271行预测完成！
[ 229.]
第272行预测完成！
[ 342.]
第273行预测完成！
[ 349.]
第274行预测完成！
[ 439.]
第275行预测完成！
[ 428.]
第276行预测完成！
[ 348.]
第277行预测完成！
[ 310.]
第278行预测完成！
[ 322.]
第279行预测完成！
[ 333.]
第280行预测完成！
[ 479.]
第281行预测完成！
[ 423.]
第282行预测完成！
[ 323.]
第283行预测完成！
[ 251.]
第284行预测完成！
[ 228.]
第285行预测完成！
[ 193.]
第286行预测完成！
[ 77.]
第287行预测完成！
[ 28.]
第288行预测完成！
[ 15.]
第289行预测完成！
[ 14.]
第290行预测完成！
[ 13.]
第291行预测完成！
[ 16.]
第292行预测完成！
[ 27.]


第682行预测完成！
[ 691.]
第683行预测完成！
[ 697.]
第684行预测完成！
[ 508.]
第685行预测完成！
[ 454.]
第686行预测完成！
[ 486.]
第687行预测完成！
[ 463.]
第688行预测完成！
[ 802.]
第689行预测完成！
[ 600.]
第690行预测完成！
[ 392.]
第691行预测完成！
[ 317.]
第692行预测完成！
[ 288.]
第693行预测完成！
[ 240.]
第694行预测完成！
[ 70.]
第695行预测完成！
[ 12.]
第696行预测完成！
[ 1.]
第697行预测完成！
[ 1.]
第698行预测完成！
[ 4.]
第699行预测完成！
[ 5.]
第700行预测完成！
[ 14.]
第701行预测完成！
[ 50.]
第702行预测完成！
[ 210.]
第703行预测完成！
[ 526.]
第704行预测完成！
[ 511.]
第705行预测完成！
[ 409.]
第706行预测完成！
[ 662.]
第707行预测完成！
[ 670.]
第708行预测完成！
[ 481.]
第709行预测完成！
[ 419.]
第710行预测完成！
[ 439.]
第711行预测完成！
[ 424.]
第712行预测完成！
[ 763.]
第713行预测完成！
[ 550.]
第714行预测完成！
[ 370.]
第715行预测完成！
[ 315.]
第716行预测完成！
[ 283.]
第717行预测完成！
[ 222.]
第718行预测完成！
[ 67.]
第719行预测完成！
******删减前地点数据大小：1440******
******删减后地点数据大小：720******
[-2.]
第0行预测完成！
[-6.]
第1行预测完成！
[ 3.]
第2行预测完成！
[ 9.]
第3行预测完成！
[ 10.]
第4行预测完成！
[ 18.]
第5行预测完成！
[ 69.]
第6行预测完成！
[ 291.]
第7行预测完成！
[ 519.]
第8行预测完成！
[ 716.]
第9行预测完成！
[ 486.]
第10行预测完成！
[ 552.]
第11行预测完成！
[ 699.]
第12行预测完成！
[ 645.]
第13行预测完成！
[ 529.]
第14行预测完

第413行预测完成！
[ 38.]
第414行预测完成！
[ 141.]
第415行预测完成！
[ 297.]
第416行预测完成！
[ 389.]
第417行预测完成！
[ 367.]
第418行预测完成！
[ 505.]
第419行预测完成！
[ 596.]
第420行预测完成！
[ 395.]
第421行预测完成！
[ 363.]
第422行预测完成！
[ 369.]
第423行预测完成！
[ 370.]
第424行预测完成！
[ 574.]
第425行预测完成！
[ 491.]
第426行预测完成！
[ 337.]
第427行预测完成！
[ 280.]
第428行预测完成！
[ 267.]
第429行预测完成！
[ 212.]
第430行预测完成！
[ 58.]
第431行预测完成！
[ 3.]
第432行预测完成！
[-8.]
第433行预测完成！
[-6.]
第434行预测完成！
[-4.]
第435行预测完成！
[-3.]
第436行预测完成！
[ 1.]
第437行预测完成！
[ 16.]
第438行预测完成！
[ 101.]
第439行预测完成！
[ 241.]
第440行预测完成！
[ 348.]
第441行预测完成！
[ 340.]
第442行预测完成！
[ 448.]
第443行预测完成！
[ 579.]
第444行预测完成！
[ 396.]
第445行预测完成！
[ 348.]
第446行预测完成！
[ 401.]
第447行预测完成！
[ 379.]
第448行预测完成！
[ 621.]
第449行预测完成！
[ 497.]
第450行预测完成！
[ 350.]
第451行预测完成！
[ 259.]
第452行预测完成！
[ 237.]
第453行预测完成！
[ 194.]
第454行预测完成！
[ 58.]
第455行预测完成！
[ 8.]
第456行预测完成！
[ 2.]
第457行预测完成！
[ 2.]
第458行预测完成！
[ 4.]
第459行预测完成！
[ 8.]
第460行预测完成！
[ 15.]
第461行预测完成！
[ 42.]
第462行预测完成！
[ 165.]
第463行预测完成！
[ 419.]
第464行预测完成！
[ 592.]
第465行预测完成！
[ 446.]
第466行预测完成！
[ 557.]
第4

第139行预测完成！
[ 341.]
第140行预测完成！
[ 277.]
第141行预测完成！
[ 200.]
第142行预测完成！
[ 68.]
第143行预测完成！
[ 6.]
第144行预测完成！
[-4.]
第145行预测完成！
[ 1.]
第146行预测完成！
[ 5.]
第147行预测完成！
[ 9.]
第148行预测完成！
[ 24.]
第149行预测完成！
[ 58.]
第150行预测完成！
[ 263.]
第151行预测完成！
[ 611.]
第152行预测完成！
[ 721.]
第153行预测完成！
[ 614.]
第154行预测完成！
[ 885.]
第155行预测完成！
[ 839.]
第156行预测完成！
[ 708.]
第157行预测完成！
[ 655.]
第158行预测完成！
[ 822.]
第159行预测完成！
[ 701.]
第160行预测完成！
[ 1125.]
第161行预测完成！
[ 722.]
第162行预测完成！
[ 453.]
第163行预测完成！
[ 367.]
第164行预测完成！
[ 302.]
第165行预测完成！
[ 202.]
第166行预测完成！
[ 57.]
第167行预测完成！
[ 9.]
第168行预测完成！
[-1.]
第169行预测完成！
[ 1.]
第170行预测完成！
[ 4.]
第171行预测完成！
[ 7.]
第172行预测完成！
[ 24.]
第173行预测完成！
[ 73.]
第174行预测完成！
[ 298.]
第175行预测完成！
[ 633.]
第176行预测完成！
[ 709.]
第177行预测完成！
[ 612.]
第178行预测完成！
[ 879.]
第179行预测完成！
[ 863.]
第180行预测完成！
[ 683.]
第181行预测完成！
[ 647.]
第182行预测完成！
[ 755.]
第183行预测完成！
[ 680.]
第184行预测完成！
[ 1089.]
第185行预测完成！
[ 737.]
第186行预测完成！
[ 449.]
第187行预测完成！
[ 384.]
第188行预测完成！
[ 296.]
第189行预测完成！
[ 198.]
第190行预测完成！
[ 53.]
第191行预测完成！
[ 5.]
第192行预测完成！
[-4.]
第19

第587行预测完成！
[ 555.]
第588行预测完成！
[ 373.]
第589行预测完成！
[ 348.]
第590行预测完成！
[ 360.]
第591行预测完成！
[ 394.]
第592行预测完成！
[ 554.]
第593行预测完成！
[ 549.]
第594行预测完成！
[ 386.]
第595行预测完成！
[ 305.]
第596行预测完成！
[ 247.]
第597行预测完成！
[ 170.]
第598行预测完成！
[ 50.]
第599行预测完成！
[-2.]
第600行预测完成！
[-8.]
第601行预测完成！
[-7.]
第602行预测完成！
[-4.]
第603行预测完成！
[-2.]
第604行预测完成！
[ 2.]
第605行预测完成！
[ 19.]
第606行预测完成！
[ 99.]
第607行预测完成！
[ 232.]
第608行预测完成！
[ 350.]
第609行预测完成！
[ 360.]
第610行预测完成！
[ 470.]
第611行预测完成！
[ 573.]
第612行预测完成！
[ 461.]
第613行预测完成！
[ 420.]
第614行预测完成！
[ 456.]
第615行预测完成！
[ 434.]
第616行预测完成！
[ 648.]
第617行预测完成！
[ 577.]
第618行预测完成！
[ 392.]
第619行预测完成！
[ 283.]
第620行预测完成！
[ 239.]
第621行预测完成！
[ 165.]
第622行预测完成！
[ 52.]
第623行预测完成！
[ 10.]
第624行预测完成！
[ 4.]
第625行预测完成！
[ 1.]
第626行预测完成！
[ 4.]
第627行预测完成！
[ 7.]
第628行预测完成！
[ 14.]
第629行预测完成！
[ 39.]
第630行预测完成！
[ 149.]
第631行预测完成！
[ 420.]
第632行预测完成！
[ 582.]
第633行预测完成！
[ 478.]
第634行预测完成！
[ 747.]
第635行预测完成！
[ 784.]
第636行预测完成！
[ 617.]
第637行预测完成！
[ 542.]
第638行预测完成！
[ 706.]
第639行预测完成！
[ 647.]
第640行预测完成！
[ 1091.]


第302行预测完成！
[ 735.]
第303行预测完成！
[ 668.]
第304行预测完成！
[ 1312.]
第305行预测完成！
[ 1357.]
第306行预测完成！
[ 1013.]
第307行预测完成！
[ 811.]
第308行预测完成！
[ 870.]
第309行预测完成！
[ 995.]
第310行预测完成！
[ 582.]
第311行预测完成！
[ 325.]
第312行预测完成！
[ 249.]
第313行预测完成！
[ 205.]
第314行预测完成！
[ 176.]
第315行预测完成！
[ 149.]
第316行预测完成！
[ 147.]
第317行预测完成！
[ 194.]
第318行预测完成！
[ 506.]
第319行预测完成！
[ 872.]
第320行预测完成！
[ 771.]
第321行预测完成！
[ 543.]
第322行预测完成！
[ 1180.]
第323行预测完成！
[ 1645.]
第324行预测完成！
[ 1128.]
第325行预测完成！
[ 886.]
第326行预测完成！
[ 720.]
第327行预测完成！
[ 665.]
第328行预测完成！
[ 1318.]
第329行预测完成！
[ 1348.]
第330行预测完成！
[ 1015.]
第331行预测完成！
[ 829.]
第332行预测完成！
[ 894.]
第333行预测完成！
[ 1051.]
第334行预测完成！
[ 611.]
第335行预测完成！
[ 367.]
第336行预测完成！
[ 276.]
第337行预测完成！
[ 227.]
第338行预测完成！
[ 184.]
第339行预测完成！
[ 158.]
第340行预测完成！
[ 152.]
第341行预测完成！
[ 190.]
第342行预测完成！
[ 504.]
第343行预测完成！
[ 886.]
第344行预测完成！
[ 761.]
第345行预测完成！
[ 529.]
第346行预测完成！
[ 1152.]
第347行预测完成！
[ 1651.]
第348行预测完成！
[ 1181.]
第349行预测完成！
[ 887.]
第350行预测完成！
[ 717.]
第351行预测完成！
[ 658.]
第352行预测完成！
[ 1267.]
第353行预测完成！
[ 1362

第7行预测完成！
[ 520.]
第8行预测完成！
[ 457.]
第9行预测完成！
[ 340.]
第10行预测完成！
[ 477.]
第11行预测完成！
[ 622.]
第12行预测完成！
[ 497.]
第13行预测完成！
[ 481.]
第14行预测完成！
[ 397.]
第15行预测完成！
[ 421.]
第16行预测完成！
[ 567.]
第17行预测完成！
[ 542.]
第18行预测完成！
[ 413.]
第19行预测完成！
[ 368.]
第20行预测完成！
[ 408.]
第21行预测完成！
[ 508.]
第22行预测完成！
[ 625.]
第23行预测完成！
[ 543.]
第24行预测完成！
[ 422.]
第25行预测完成！
[ 359.]
第26行预测完成！
[ 368.]
第27行预测完成！
[ 320.]
第28行预测完成！
[ 284.]
第29行预测完成！
[ 304.]
第30行预测完成！
[ 421.]
第31行预测完成！
[ 569.]
第32行预测完成！
[ 451.]
第33行预测完成！
[ 335.]
第34行预测完成！
[ 515.]
第35行预测完成！
[ 621.]
第36行预测完成！
[ 473.]
第37行预测完成！
[ 428.]
第38行预测完成！
[ 351.]
第39行预测完成！
[ 379.]
第40行预测完成！
[ 533.]
第41行预测完成！
[ 538.]
第42行预测完成！
[ 404.]
第43行预测完成！
[ 361.]
第44行预测完成！
[ 385.]
第45行预测完成！
[ 439.]
第46行预测完成！
[ 422.]
第47行预测完成！
[ 420.]
第48行预测完成！
[ 356.]
第49行预测完成！
[ 329.]
第50行预测完成！
[ 318.]
第51行预测完成！
[ 295.]
第52行预测完成！
[ 284.]
第53行预测完成！
[ 293.]
第54行预测完成！
[ 398.]
第55行预测完成！
[ 530.]
第56行预测完成！
[ 436.]
第57行预测完成！
[ 347.]
第58行预测完成！
[ 536.]
第59行预测完成！
[ 644.]
第60行预测完成！
[ 484.]
第61行预测完成！
[ 435.]
第62行预测完成！
[ 3

第446行预测完成！
[ 296.]
第447行预测完成！
[ 305.]
第448行预测完成！
[ 414.]
第449行预测完成！
[ 387.]
第450行预测完成！
[ 297.]
第451行预测完成！
[ 246.]
第452行预测完成！
[ 271.]
第453行预测完成！
[ 281.]
第454行预测完成！
[ 246.]
第455行预测完成！
[ 239.]
第456行预测完成！
[ 255.]
第457行预测完成！
[ 268.]
第458行预测完成！
[ 283.]
第459行预测完成！
[ 295.]
第460行预测完成！
[ 301.]
第461行预测完成！
[ 316.]
第462行预测完成！
[ 384.]
第463行预测完成！
[ 459.]
第464行预测完成！
[ 389.]
第465行预测完成！
[ 333.]
第466行预测完成！
[ 483.]
第467行预测完成！
[ 592.]
第468行预测完成！
[ 443.]
第469行预测完成！
[ 378.]
第470行预测完成！
[ 332.]
第471行预测完成！
[ 327.]
第472行预测完成！
[ 492.]
第473行预测完成！
[ 430.]
第474行预测完成！
[ 325.]
第475行预测完成！
[ 291.]
第476行预测完成！
[ 332.]
第477行预测完成！
[ 369.]
第478行预测完成！
[ 361.]
第479行预测完成！
[ 309.]
第480行预测完成！
[ 287.]
第481行预测完成！
[ 303.]
第482行预测完成！
[ 306.]
第483行预测完成！
[ 296.]
第484行预测完成！
[ 312.]
第485行预测完成！
[ 315.]
第486行预测完成！
[ 396.]
第487行预测完成！
[ 494.]
第488行预测完成！
[ 402.]
第489行预测完成！
[ 335.]
第490行预测完成！
[ 513.]
第491行预测完成！
[ 642.]
第492行预测完成！
[ 473.]
第493行预测完成！
[ 378.]
第494行预测完成！
[ 333.]
第495行预测完成！
[ 334.]
第496行预测完成！
[ 535.]
第497行预测完成！
[ 471.]
第498行预测完成！
[

第162行预测完成！
[ 594.]
第163行预测完成！
[ 482.]
第164行预测完成！
[ 515.]
第165行预测完成！
[ 635.]
第166行预测完成！
[ 401.]
第167行预测完成！
[ 238.]
第168行预测完成！
[ 181.]
第169行预测完成！
[ 159.]
第170行预测完成！
[ 144.]
第171行预测完成！
[ 130.]
第172行预测完成！
[ 125.]
第173行预测完成！
[ 169.]
第174行预测完成！
[ 514.]
第175行预测完成！
[ 513.]
第176行预测完成！
[ 464.]
第177行预测完成！
[ 381.]
第178行预测完成！
[ 640.]
第179行预测完成！
[ 1085.]
第180行预测完成！
[ 794.]
第181行预测完成！
[ 573.]
第182行预测完成！
[ 532.]
第183行预测完成！
[ 455.]
第184行预测完成！
[ 902.]
第185行预测完成！
[ 862.]
第186行预测完成！
[ 573.]
第187行预测完成！
[ 455.]
第188行预测完成！
[ 500.]
第189行预测完成！
[ 622.]
第190行预测完成！
[ 397.]
第191行预测完成！
[ 238.]
第192行预测完成！
[ 187.]
第193行预测完成！
[ 155.]
第194行预测完成！
[ 142.]
第195行预测完成！
[ 126.]
第196行预测完成！
[ 122.]
第197行预测完成！
[ 168.]
第198行预测完成！
[ 475.]
第199行预测完成！
[ 502.]
第200行预测完成！
[ 470.]
第201行预测完成！
[ 374.]
第202行预测完成！
[ 663.]
第203行预测完成！
[ 1031.]
第204行预测完成！
[ 756.]
第205行预测完成！
[ 530.]
第206行预测完成！
[ 495.]
第207行预测完成！
[ 446.]
第208行预测完成！
[ 881.]
第209行预测完成！
[ 828.]
第210行预测完成！
[ 568.]
第211行预测完成！
[ 444.]
第212行预测完成！
[ 470.]
第213行预测完成！
[ 582.]
第214行预测完成！

第595行预测完成！
[ 374.]
第596行预测完成！
[ 381.]
第597行预测完成！
[ 410.]
第598行预测完成！
[ 293.]
第599行预测完成！
[ 172.]
第600行预测完成！
[ 138.]
第601行预测完成！
[ 121.]
第602行预测完成！
[ 101.]
第603行预测完成！
[ 93.]
第604行预测完成！
[ 89.]
第605行预测完成！
[ 115.]
第606行预测完成！
[ 197.]
第607行预测完成！
[ 318.]
第608行预测完成！
[ 347.]
第609行预测完成！
[ 304.]
第610行预测完成！
[ 512.]
第611行预测完成！
[ 894.]
第612行预测完成！
[ 597.]
第613行预测完成！
[ 422.]
第614行预测完成！
[ 374.]
第615行预测完成！
[ 355.]
第616行预测完成！
[ 673.]
第617行预测完成！
[ 693.]
第618行预测完成！
[ 503.]
第619行预测完成！
[ 360.]
第620行预测完成！
[ 346.]
第621行预测完成！
[ 402.]
第622行预测完成！
[ 277.]
第623行预测完成！
[ 170.]
第624行预测完成！
[ 148.]
第625行预测完成！
[ 132.]
第626行预测完成！
[ 110.]
第627行预测完成！
[ 104.]
第628行预测完成！
[ 106.]
第629行预测完成！
[ 138.]
第630行预测完成！
[ 293.]
第631行预测完成！
[ 484.]
第632行预测完成！
[ 413.]
第633行预测完成！
[ 340.]
第634行预测完成！
[ 643.]
第635行预测完成！
[ 1024.]
第636行预测完成！
[ 720.]
第637行预测完成！
[ 528.]
第638行预测完成！
[ 452.]
第639行预测完成！
[ 414.]
第640行预测完成！
[ 868.]
第641行预测完成！
[ 839.]
第642行预测完成！
[ 571.]
第643行预测完成！
[ 439.]
第644行预测完成！
[ 458.]
第645行预测完成！
[ 540.]
第646行预测完成！
[ 361.]
第647行预测完成！
[ 

第312行预测完成！
[-2.]
第313行预测完成！
[-8.]
第314行预测完成！
[-2.]
第315行预测完成！
[ 3.]
第316行预测完成！
[ 14.]
第317行预测完成！
[ 82.]
第318行预测完成！
[ 420.]
第319行预测完成！
[ 1594.]
第320行预测完成！
[ 1742.]
第321行预测完成！
[ 1513.]
第322行预测完成！
[ 2045.]
第323行预测完成！
[ 2084.]
第324行预测完成！
[ 1785.]
第325行预测完成！
[ 2295.]
第326行预测完成！
[ 2064.]
第327行预测完成！
[ 2025.]
第328行预测完成！
[ 2410.]
第329行预测完成！
[ 2063.]
第330行预测完成！
[ 1529.]
第331行预测完成！
[ 1153.]
第332行预测完成！
[ 915.]
第333行预测完成！
[ 739.]
第334行预测完成！
[ 242.]
第335行预测完成！
[ 29.]
第336行预测完成！
[ 10.]
第337行预测完成！
[-5.]
第338行预测完成！
[-4.]
第339行预测完成！
[ 5.]
第340行预测完成！
[ 15.]
第341行预测完成！
[ 100.]
第342行预测完成！
[ 496.]
第343行预测完成！
[ 1634.]
第344行预测完成！
[ 1761.]
第345行预测完成！
[ 1541.]
第346行预测完成！
[ 2031.]
第347行预测完成！
[ 2185.]
第348行预测完成！
[ 1846.]
第349行预测完成！
[ 2136.]
第350行预测完成！
[ 2050.]
第351行预测完成！
[ 2035.]
第352行预测完成！
[ 2462.]
第353行预测完成！
[ 2072.]
第354行预测完成！
[ 1499.]
第355行预测完成！
[ 1134.]
第356行预测完成！
[ 906.]
第357行预测完成！
[ 685.]
第358行预测完成！
[ 229.]
第359行预测完成！
[ 30.]
第360行预测完成！
[ 8.]
第361行预测完成！
[ 5.]
第362行预测完成！
[ 5.]
第363行预测完成！
[ 13.]
第364行预测完成！
[ 

第20行预测完成！
[ 689.]
第21行预测完成！
[ 804.]
第22行预测完成！
[ 614.]
第23行预测完成！
[ 613.]
第24行预测完成！
[ 546.]
第25行预测完成！
[ 432.]
第26行预测完成！
[ 408.]
第27行预测完成！
[ 421.]
第28行预测完成！
[ 430.]
第29行预测完成！
[ 355.]
第30行预测完成！
[ 381.]
第31行预测完成！
[ 669.]
第32行预测完成！
[ 646.]
第33行预测完成！
[ 481.]
第34行预测完成！
[ 854.]
第35行预测完成！
[ 1274.]
第36行预测完成！
[ 845.]
第37行预测完成！
[ 712.]
第38行预测完成！
[ 605.]
第39行预测完成！
[ 532.]
第40行预测完成！
[ 977.]
第41行预测完成！
[ 1113.]
第42行预测完成！
[ 800.]
第43行预测完成！
[ 671.]
第44行预测完成！
[ 689.]
第45行预测完成！
[ 756.]
第46行预测完成！
[ 541.]
第47行预测完成！
[ 514.]
第48行预测完成！
[ 500.]
第49行预测完成！
[ 413.]
第50行预测完成！
[ 404.]
第51行预测完成！
[ 397.]
第52行预测完成！
[ 398.]
第53行预测完成！
[ 333.]
第54行预测完成！
[ 371.]
第55行预测完成！
[ 572.]
第56行预测完成！
[ 615.]
第57行预测完成！
[ 483.]
第58行预测完成！
[ 842.]
第59行预测完成！
[ 1249.]
第60行预测完成！
[ 825.]
第61行预测完成！
[ 684.]
第62行预测完成！
[ 603.]
第63行预测完成！
[ 533.]
第64行预测完成！
[ 947.]
第65行预测完成！
[ 1012.]
第66行预测完成！
[ 720.]
第67行预测完成！
[ 616.]
第68行预测完成！
[ 639.]
第69行预测完成！
[ 705.]
第70行预测完成！
[ 501.]
第71行预测完成！
[ 397.]
第72行预测完成！
[ 426.]
第73行预测完成！
[ 364.]
第74行预测完成！
[ 352.]
第75行预测

第456行预测完成！
[ 397.]
第457行预测完成！
[ 371.]
第458行预测完成！
[ 370.]
第459行预测完成！
[ 392.]
第460行预测完成！
[ 400.]
第461行预测完成！
[ 401.]
第462行预测完成！
[ 426.]
第463行预测完成！
[ 596.]
第464行预测完成！
[ 583.]
第465行预测完成！
[ 468.]
第466行预测完成！
[ 800.]
第467行预测完成！
[ 1164.]
第468行预测完成！
[ 863.]
第469行预测完成！
[ 694.]
第470行预测完成！
[ 617.]
第471行预测完成！
[ 538.]
第472行预测完成！
[ 1050.]
第473行预测完成！
[ 1144.]
第474行预测完成！
[ 834.]
第475行预测完成！
[ 651.]
第476行预测完成！
[ 654.]
第477行预测完成！
[ 744.]
第478行预测完成！
[ 516.]
第479行预测完成！
[ 465.]
第480行预测完成！
[ 419.]
第481行预测完成！
[ 374.]
第482行预测完成！
[ 360.]
第483行预测完成！
[ 373.]
第484行预测完成！
[ 389.]
第485行预测完成！
[ 392.]
第486行预测完成！
[ 449.]
第487行预测完成！
[ 654.]
第488行预测完成！
[ 601.]
第489行预测完成！
[ 473.]
第490行预测完成！
[ 786.]
第491行预测完成！
[ 1189.]
第492行预测完成！
[ 874.]
第493行预测完成！
[ 706.]
第494行预测完成！
[ 606.]
第495行预测完成！
[ 530.]
第496行预测完成！
[ 1025.]
第497行预测完成！
[ 1119.]
第498行预测完成！
[ 793.]
第499行预测完成！
[ 641.]
第500行预测完成！
[ 675.]
第501行预测完成！
[ 738.]
第502行预测完成！
[ 515.]
第503行预测完成！
[ 468.]
第504行预测完成！
[ 420.]
第505行预测完成！
[ 373.]
第506行预测完成！
[ 353.]
第507行预测完成！
[ 361.]
第508行预

第178行预测完成！
[ 379.]
第179行预测完成！
[ 385.]
第180行预测完成！
[ 314.]
第181行预测完成！
[ 282.]
第182行预测完成！
[ 384.]
第183行预测完成！
[ 327.]
第184行预测完成！
[ 525.]
第185行预测完成！
[ 402.]
第186行预测完成！
[ 280.]
第187行预测完成！
[ 222.]
第188行预测完成！
[ 167.]
第189行预测完成！
[ 115.]
第190行预测完成！
[ 30.]
第191行预测完成！
[ 2.]
第192行预测完成！
[-4.]
第193行预测完成！
[-1.]
第194行预测完成！
[ 0.]
第195行预测完成！
[ 4.]
第196行预测完成！
[ 8.]
第197行预测完成！
[ 28.]
第198行预测完成！
[ 126.]
第199行预测完成！
[ 274.]
第200行预测完成！
[ 441.]
第201行预测完成！
[ 328.]
第202行预测完成！
[ 367.]
第203行预测完成！
[ 389.]
第204行预测完成！
[ 334.]
第205行预测完成！
[ 282.]
第206行预测完成！
[ 374.]
第207行预测完成！
[ 316.]
第208行预测完成！
[ 484.]
第209行预测完成！
[ 384.]
第210行预测完成！
[ 270.]
第211行预测完成！
[ 201.]
第212行预测完成！
[ 152.]
第213行预测完成！
[ 105.]
第214行预测完成！
[ 43.]
第215行预测完成！
[ 7.]
第216行预测完成！
[-2.]
第217行预测完成！
[-2.]
第218行预测完成！
[-1.]
第219行预测完成！
[ 3.]
第220行预测完成！
[ 6.]
第221行预测完成！
[ 23.]
第222行预测完成！
[ 105.]
第223行预测完成！
[ 244.]
第224行预测完成！
[ 398.]
第225行预测完成！
[ 304.]
第226行预测完成！
[ 340.]
第227行预测完成！
[ 377.]
第228行预测完成！
[ 311.]
第229行预测完成！
[ 262.]
第230行预测完成！
[ 320.]
第231行预测完成！
[ 283.]
第2

第626行预测完成！
[ 4.]
第627行预测完成！
[ 6.]
第628行预测完成！
[ 11.]
第629行预测完成！
[ 29.]
第630行预测完成！
[ 105.]
第631行预测完成！
[ 235.]
第632行预测完成！
[ 324.]
第633行预测完成！
[ 261.]
第634行预测完成！
[ 328.]
第635行预测完成！
[ 346.]
第636行预测完成！
[ 308.]
第637行预测完成！
[ 262.]
第638行预测完成！
[ 290.]
第639行预测完成！
[ 278.]
第640行预测完成！
[ 447.]
第641行预测完成！
[ 355.]
第642行预测完成！
[ 237.]
第643行预测完成！
[ 166.]
第644行预测完成！
[ 128.]
第645行预测完成！
[ 88.]
第646行预测完成！
[ 29.]
第647行预测完成！
[ 8.]
第648行预测完成！
[ 3.]
第649行预测完成！
[ 2.]
第650行预测完成！
[ 3.]
第651行预测完成！
[ 7.]
第652行预测完成！
[ 11.]
第653行预测完成！
[ 33.]
第654行预测完成！
[ 120.]
第655行预测完成！
[ 252.]
第656行预测完成！
[ 355.]
第657行预测完成！
[ 286.]
第658行预测完成！
[ 351.]
第659行预测完成！
[ 346.]
第660行预测完成！
[ 282.]
第661行预测完成！
[ 251.]
第662行预测完成！
[ 268.]
第663行预测完成！
[ 265.]
第664行预测完成！
[ 456.]
第665行预测完成！
[ 357.]
第666行预测完成！
[ 241.]
第667行预测完成！
[ 168.]
第668行预测完成！
[ 122.]
第669行预测完成！
[ 78.]
第670行预测完成！
[ 22.]
第671行预测完成！
[-0.]
第672行预测完成！
[-2.]
第673行预测完成！
[ 0.]
第674行预测完成！
[ 2.]
第675行预测完成！
[ 4.]
第676行预测完成！
[ 8.]
第677行预测完成！
[ 28.]
第678行预测完成！
[ 103.]
第679行预测完成！
[ 251.]
第680行预测完成

第345行预测完成！
[ 330.]
第346行预测完成！
[ 534.]
第347行预测完成！
[ 633.]
第348行预测完成！
[ 505.]
第349行预测完成！
[ 399.]
第350行预测完成！
[ 355.]
第351行预测完成！
[ 343.]
第352行预测完成！
[ 597.]
第353行预测完成！
[ 504.]
第354行预测完成！
[ 354.]
第355行预测完成！
[ 315.]
第356行预测完成！
[ 350.]
第357行预测完成！
[ 366.]
第358行预测完成！
[ 439.]
第359行预测完成！
[ 415.]
第360行预测完成！
[ 380.]
第361行预测完成！
[ 357.]
第362行预测完成！
[ 346.]
第363行预测完成！
[ 361.]
第364行预测完成！
[ 370.]
第365行预测完成！
[ 382.]
第366行预测完成！
[ 485.]
第367行预测完成！
[ 441.]
第368行预测完成！
[ 368.]
第369行预测完成！
[ 324.]
第370行预测完成！
[ 505.]
第371行预测完成！
[ 626.]
第372行预测完成！
[ 493.]
第373行预测完成！
[ 387.]
第374行预测完成！
[ 347.]
第375行预测完成！
[ 329.]
第376行预测完成！
[ 543.]
第377行预测完成！
[ 461.]
第378行预测完成！
[ 338.]
第379行预测完成！
[ 303.]
第380行预测完成！
[ 354.]
第381行预测完成！
[ 350.]
第382行预测完成！
[ 367.]
第383行预测完成！
[ 376.]
第384行预测完成！
[ 364.]
第385行预测完成！
[ 352.]
第386行预测完成！
[ 344.]
第387行预测完成！
[ 354.]
第388行预测完成！
[ 361.]
第389行预测完成！
[ 372.]
第390行预测完成！
[ 452.]
第391行预测完成！
[ 427.]
第392行预测完成！
[ 363.]
第393行预测完成！
[ 320.]
第394行预测完成！
[ 474.]
第395行预测完成！
[ 624.]
第396行预测完成！
[ 472.]
第397行预测完成！
[

[ 1394.]
第59行预测完成！
[ 1306.]
第60行预测完成！
[ 1128.]
第61行预测完成！
[ 937.]
第62行预测完成！
[ 1060.]
第63行预测完成！
[ 960.]
第64行预测完成！
[ 1574.]
第65行预测完成！
[ 1201.]
第66行预测完成！
[ 779.]
第67行预测完成！
[ 594.]
第68行预测完成！
[ 575.]
第69行预测完成！
[ 469.]
第70行预测完成！
[ 178.]
第71行预测完成！
[ 49.]
第72行预测完成！
[ 7.]
第73行预测完成！
[-5.]
第74行预测完成！
[-6.]
第75行预测完成！
[-1.]
第76行预测完成！
[ 14.]
第77行预测完成！
[ 94.]
第78行预测完成！
[ 298.]
第79行预测完成！
[ 499.]
第80行预测完成！
[ 776.]
第81行预测完成！
[ 754.]
第82行预测完成！
[ 1175.]
第83行预测完成！
[ 1097.]
第84行预测完成！
[ 758.]
第85行预测完成！
[ 711.]
第86行预测完成！
[ 760.]
第87行预测完成！
[ 846.]
第88行预测完成！
[ 1096.]
第89行预测完成！
[ 939.]
第90行预测完成！
[ 609.]
第91行预测完成！
[ 479.]
第92行预测完成！
[ 438.]
第93行预测完成！
[ 377.]
第94行预测完成！
[ 147.]
第95行预测完成！
[ 47.]
第96行预测完成！
[ 14.]
第97行预测完成！
[ 2.]
第98行预测完成！
[-2.]
第99行预测完成！
[ 1.]
第100行预测完成！
[ 10.]
第101行预测完成！
[ 71.]
第102行预测完成！
[ 236.]
第103行预测完成！
[ 444.]
第104行预测完成！
[ 660.]
第105行预测完成！
[ 873.]
第106行预测完成！
[ 1165.]
第107行预测完成！
[ 1068.]
第108行预测完成！
[ 858.]
第109行预测完成！
[ 780.]
第110行预测完成！
[ 815.]
第111行预测完成！
[ 817.]
第112行预测完成！
[ 1235.]
第113行预测完成！
[ 999

第497行预测完成！
[ 1461.]
第498行预测完成！
[ 923.]
第499行预测完成！
[ 681.]
第500行预测完成！
[ 602.]
第501行预测完成！
[ 468.]
第502行预测完成！
[ 178.]
第503行预测完成！
[ 40.]
第504行预测完成！
[ 6.]
第505行预测完成！
[ 1.]
第506行预测完成！
[ 0.]
第507行预测完成！
[ 5.]
第508行预测完成！
[ 23.]
第509行预测完成！
[ 105.]
第510行预测完成！
[ 531.]
第511行预测完成！
[ 1033.]
第512行预测完成！
[ 950.]
第513行预测完成！
[ 875.]
第514行预测完成！
[ 1568.]
第515行预测完成！
[ 1473.]
第516行预测完成！
[ 1061.]
第517行预测完成！
[ 954.]
第518行预测完成！
[ 1018.]
第519行预测完成！
[ 1049.]
第520行预测完成！
[ 1768.]
第521行预测完成！
[ 1459.]
第522行预测完成！
[ 916.]
第523行预测完成！
[ 685.]
第524行预测完成！
[ 585.]
第525行预测完成！
[ 408.]
第526行预测完成！
[ 155.]
第527行预测完成！
[ 41.]
第528行预测完成！
[ 17.]
第529行预测完成！
[ 14.]
第530行预测完成！
[ 12.]
第531行预测完成！
[ 12.]
第532行预测完成！
[ 25.]
第533行预测完成！
[ 105.]
第534行预测完成！
[ 509.]
第535行预测完成！
[ 1021.]
第536行预测完成！
[ 941.]
第537行预测完成！
[ 869.]
第538行预测完成！
[ 1608.]
第539行预测完成！
[ 1508.]
第540行预测完成！
[ 1046.]
第541行预测完成！
[ 920.]
第542行预测完成！
[ 966.]
第543行预测完成！
[ 1014.]
第544行预测完成！
[ 1744.]
第545行预测完成！
[ 1438.]
第546行预测完成！
[ 914.]
第547行预测完成！
[ 662.]
第548行预测完成！
[ 538.]
第549行预测完成！
[

第211行预测完成！
[ 754.]
第212行预测完成！
[ 754.]
第213行预测完成！
[ 843.]
第214行预测完成！
[ 430.]
第215行预测完成！
[ 388.]
第216行预测完成！
[ 479.]
第217行预测完成！
[ 486.]
第218行预测完成！
[ 460.]
第219行预测完成！
[ 453.]
第220行预测完成！
[ 465.]
第221行预测完成！
[ 453.]
第222行预测完成！
[ 488.]
第223行预测完成！
[ 695.]
第224行预测完成！
[ 700.]
第225行预测完成！
[ 599.]
第226行预测完成！
[ 1041.]
第227行预测完成！
[ 1587.]
第228行预测完成！
[ 1082.]
第229行预测完成！
[ 863.]
第230行预测完成！
[ 725.]
第231行预测完成！
[ 725.]
第232行预测完成！
[ 1205.]
第233行预测完成！
[ 1338.]
第234行预测完成！
[ 961.]
第235行预测完成！
[ 730.]
第236行预测完成！
[ 752.]
第237行预测完成！
[ 808.]
第238行预测完成！
[ 425.]
第239行预测完成！
[ 320.]
第240行预测完成！
[ 440.]
第241行预测完成！
[ 423.]
第242行预测完成！
[ 417.]
第243行预测完成！
[ 419.]
第244行预测完成！
[ 427.]
第245行预测完成！
[ 421.]
第246行预测完成！
[ 413.]
第247行预测完成！
[ 505.]
第248行预测完成！
[ 590.]
第249行预测完成！
[ 537.]
第250行预测完成！
[ 951.]
第251行预测完成！
[ 1357.]
第252行预测完成！
[ 957.]
第253行预测完成！
[ 746.]
第254行预测完成！
[ 649.]
第255行预测完成！
[ 646.]
第256行预测完成！
[ 1042.]
第257行预测完成！
[ 1214.]
第258行预测完成！
[ 870.]
第259行预测完成！
[ 670.]
第260行预测完成！
[ 680.]
第261行预测完成！
[ 700.]
第262行预测完成！
[ 371.]
第263

第639行预测完成！
[ 728.]
第640行预测完成！
[ 1292.]
第641行预测完成！
[ 1307.]
第642行预测完成！
[ 980.]
第643行预测完成！
[ 772.]
第644行预测完成！
[ 766.]
第645行预测完成！
[ 804.]
第646行预测完成！
[ 454.]
第647行预测完成！
[ 368.]
第648行预测完成！
[ 384.]
第649行预测完成！
[ 416.]
第650行预测完成！
[ 428.]
第651行预测完成！
[ 425.]
第652行预测完成！
[ 459.]
第653行预测完成！
[ 486.]
第654行预测完成！
[ 615.]
第655行预测完成！
[ 781.]
第656行预测完成！
[ 699.]
第657行预测完成！
[ 542.]
第658行预测完成！
[ 1029.]
第659行预测完成！
[ 1391.]
第660行预测完成！
[ 1059.]
第661行预测完成！
[ 885.]
第662行预测完成！
[ 743.]
第663行预测完成！
[ 712.]
第664行预测完成！
[ 1259.]
第665行预测完成！
[ 1292.]
第666行预测完成！
[ 993.]
第667行预测完成！
[ 768.]
第668行预测完成！
[ 768.]
第669行预测完成！
[ 810.]
第670行预测完成！
[ 450.]
第671行预测完成！
[ 352.]
第672行预测完成！
[ 378.]
第673行预测完成！
[ 436.]
第674行预测完成！
[ 428.]
第675行预测完成！
[ 412.]
第676行预测完成！
[ 445.]
第677行预测完成！
[ 475.]
第678行预测完成！
[ 635.]
第679行预测完成！
[ 764.]
第680行预测完成！
[ 674.]
第681行预测完成！
[ 530.]
第682行预测完成！
[ 987.]
第683行预测完成！
[ 1355.]
第684行预测完成！
[ 1055.]
第685行预测完成！
[ 868.]
第686行预测完成！
[ 720.]
第687行预测完成！
[ 703.]
第688行预测完成！
[ 1220.]
第689行预测完成！
[ 1251.]
第690行预测完成！
[ 956.]
第

第361行预测完成！
[ 7.]
第362行预测完成！
[ 5.]
第363行预测完成！
[ 7.]
第364行预测完成！
[ 11.]
第365行预测完成！
[ 44.]
第366行预测完成！
[ 209.]
第367行预测完成！
[ 580.]
第368行预测完成！
[ 462.]
第369行预测完成！
[ 351.]
第370行预测完成！
[ 626.]
第371行预测完成！
[ 911.]
第372行预测完成！
[ 585.]
第373行预测完成！
[ 434.]
第374行预测完成！
[ 368.]
第375行预测完成！
[ 376.]
第376行预测完成！
[ 787.]
第377行预测完成！
[ 706.]
第378行预测完成！
[ 455.]
第379行预测完成！
[ 348.]
第380行预测完成！
[ 353.]
第381行预测完成！
[ 330.]
第382行预测完成！
[ 132.]
第383行预测完成！
[ 37.]
第384行预测完成！
[ 8.]
第385行预测完成！
[ 3.]
第386行预测完成！
[ 2.]
第387行预测完成！
[ 4.]
第388行预测完成！
[ 8.]
第389行预测完成！
[ 36.]
第390行预测完成！
[ 172.]
第391行预测完成！
[ 468.]
第392行预测完成！
[ 430.]
第393行预测完成！
[ 342.]
第394行预测完成！
[ 595.]
第395行预测完成！
[ 856.]
第396行预测完成！
[ 549.]
第397行预测完成！
[ 414.]
第398行预测完成！
[ 360.]
第399行预测完成！
[ 365.]
第400行预测完成！
[ 743.]
第401行预测完成！
[ 649.]
第402行预测完成！
[ 433.]
第403行预测完成！
[ 344.]
第404行预测完成！
[ 335.]
第405行预测完成！
[ 300.]
第406行预测完成！
[ 120.]
第407行预测完成！
[ 25.]
第408行预测完成！
[-4.]
第409行预测完成！
[-6.]
第410行预测完成！
[-5.]
第411行预测完成！
[-2.]
第412行预测完成！
[ 1.]
第413行预测完成！
[ 22.]
第414行预测完成！
[ 119.]
第415行预

第86行预测完成！
[ 298.]
第87行预测完成！
[ 357.]
第88行预测完成！
[ 445.]
第89行预测完成！
[ 422.]
第90行预测完成！
[ 326.]
第91行预测完成！
[ 292.]
第92行预测完成！
[ 246.]
第93行预测完成！
[ 166.]
第94行预测完成！
[ 56.]
第95行预测完成！
[ 11.]
第96行预测完成！
[-4.]
第97行预测完成！
[-6.]
第98行预测完成！
[-4.]
第99行预测完成！
[-1.]
第100行预测完成！
[ 7.]
第101行预测完成！
[ 34.]
第102行预测完成！
[ 107.]
第103行预测完成！
[ 273.]
第104行预测完成！
[ 440.]
第105行预测完成！
[ 473.]
第106行预测完成！
[ 490.]
第107行预测完成！
[ 463.]
第108行预测完成！
[ 421.]
第109行预测完成！
[ 404.]
第110行预测完成！
[ 499.]
第111行预测完成！
[ 514.]
第112行预测完成！
[ 627.]
第113行预测完成！
[ 471.]
第114行预测完成！
[ 340.]
第115行预测完成！
[ 275.]
第116行预测完成！
[ 234.]
第117行预测完成！
[ 174.]
第118行预测完成！
[ 49.]
第119行预测完成！
[ 9.]
第120行预测完成！
[ 2.]
第121行预测完成！
[ 2.]
第122行预测完成！
[ 6.]
第123行预测完成！
[ 12.]
第124行预测完成！
[ 21.]
第125行预测完成！
[ 52.]
第126行预测完成！
[ 188.]
第127行预测完成！
[ 515.]
第128行预测完成！
[ 935.]
第129行预测完成！
[ 707.]
第130行预测完成！
[ 787.]
第131行预测完成！
[ 722.]
第132行预测完成！
[ 749.]
第133行预测完成！
[ 767.]
第134行预测完成！
[ 847.]
第135行预测完成！
[ 728.]
第136行预测完成！
[ 1099.]
第137行预测完成！
[ 753.]
第138行预测完成！
[ 483.]
第139行预测完成！
[ 451.]
第140行预测完成！
[

第532行预测完成！
[ 26.]
第533行预测完成！
[ 87.]
第534行预测完成！
[ 332.]
第535行预测完成！
[ 696.]
第536行预测完成！
[ 847.]
第537行预测完成！
[ 661.]
第538行预测完成！
[ 839.]
第539行预测完成！
[ 763.]
第540行预测完成！
[ 570.]
第541行预测完成！
[ 648.]
第542行预测完成！
[ 766.]
第543行预测完成！
[ 706.]
第544行预测完成！
[ 1019.]
第545行预测完成！
[ 743.]
第546行预测完成！
[ 467.]
第547行预测完成！
[ 375.]
第548行预测完成！
[ 309.]
第549行预测完成！
[ 200.]
第550行预测完成！
[ 59.]
第551行预测完成！
[ 8.]
第552行预测完成！
[-2.]
第553行预测完成！
[ 2.]
第554行预测完成！
[ 3.]
第555行预测完成！
[ 9.]
第556行预测完成！
[ 26.]
第557行预测完成！
[ 84.]
第558行预测完成！
[ 315.]
第559行预测完成！
[ 591.]
第560行预测完成！
[ 742.]
第561行预测完成！
[ 579.]
第562行预测完成！
[ 771.]
第563行预测完成！
[ 693.]
第564行预测完成！
[ 507.]
第565行预测完成！
[ 555.]
第566行预测完成！
[ 688.]
第567行预测完成！
[ 650.]
第568行预测完成！
[ 910.]
第569行预测完成！
[ 659.]
第570行预测完成！
[ 444.]
第571行预测完成！
[ 352.]
第572行预测完成！
[ 286.]
第573行预测完成！
[ 183.]
第574行预测完成！
[ 45.]
第575行预测完成！
[-7.]
第576行预测完成！
[-9.]
第577行预测完成！
[-5.]
第578行预测完成！
[-4.]
第579行预测完成！
[ 0.]
第580行预测完成！
[ 10.]
第581行预测完成！
[ 52.]
第582行预测完成！
[ 176.]
第583行预测完成！
[ 346.]
第584行预测完成！
[ 436.]
第585行预测完成！
[ 424.]
第

第252行预测完成！
[ 804.]
第253行预测完成！
[ 611.]
第254行预测完成！
[ 507.]
第255行预测完成！
[ 486.]
第256行预测完成！
[ 947.]
第257行预测完成！
[ 1009.]
第258行预测完成！
[ 709.]
第259行预测完成！
[ 558.]
第260行预测完成！
[ 598.]
第261行预测完成！
[ 671.]
第262行预测完成！
[ 385.]
第263行预测完成！
[ 222.]
第264行预测完成！
[ 175.]
第265行预测完成！
[ 134.]
第266行预测完成！
[ 88.]
第267行预测完成！
[ 60.]
第268行预测完成！
[ 51.]
第269行预测完成！
[ 90.]
第270行预测完成！
[ 198.]
第271行预测完成！
[ 459.]
第272行预测完成！
[ 526.]
第273行预测完成！
[ 459.]
第274行预测完成！
[ 881.]
第275行预测完成！
[ 1285.]
第276行预测完成！
[ 832.]
第277行预测完成！
[ 624.]
第278行预测完成！
[ 526.]
第279行预测完成！
[ 499.]
第280行预测完成！
[ 993.]
第281行预测完成！
[ 1069.]
第282行预测完成！
[ 788.]
第283行预测完成！
[ 573.]
第284行预测完成！
[ 566.]
第285行预测完成！
[ 611.]
第286行预测完成！
[ 367.]
第287行预测完成！
[ 238.]
第288行预测完成！
[ 176.]
第289行预测完成！
[ 136.]
第290行预测完成！
[ 99.]
第291行预测完成！
[ 74.]
第292行预测完成！
[ 58.]
第293行预测完成！
[ 96.]
第294行预测完成！
[ 246.]
第295行预测完成！
[ 677.]
第296行预测完成！
[ 665.]
第297行预测完成！
[ 499.]
第298行预测完成！
[ 970.]
第299行预测完成！
[ 1432.]
第300行预测完成！
[ 920.]
第301行预测完成！
[ 726.]
第302行预测完成！
[ 632.]
第303行预测完成！
[ 535.]
第304行预测完成！
[ 113

第685行预测完成！
[ 708.]
第686行预测完成！
[ 592.]
第687行预测完成！
[ 516.]
第688行预测完成！
[ 1135.]
第689行预测完成！
[ 1214.]
第690行预测完成！
[ 864.]
第691行预测完成！
[ 683.]
第692行预测完成！
[ 688.]
第693行预测完成！
[ 722.]
第694行预测完成！
[ 408.]
第695行预测完成！
[ 244.]
第696行预测完成！
[ 168.]
第697行预测完成！
[ 137.]
第698行预测完成！
[ 95.]
第699行预测完成！
[ 62.]
第700行预测完成！
[ 48.]
第701行预测完成！
[ 92.]
第702行预测完成！
[ 322.]
第703行预测完成！
[ 725.]
第704行预测完成！
[ 671.]
第705行预测完成！
[ 481.]
第706行预测完成！
[ 942.]
第707行预测完成！
[ 1441.]
第708行预测完成！
[ 904.]
第709行预测完成！
[ 710.]
第710行预测完成！
[ 573.]
第711行预测完成！
[ 516.]
第712行预测完成！
[ 1079.]
第713行预测完成！
[ 1171.]
第714行预测完成！
[ 855.]
第715行预测完成！
[ 669.]
第716行预测完成！
[ 666.]
第717行预测完成！
[ 699.]
第718行预测完成！
[ 403.]
第719行预测完成！
******删减前地点数据大小：1440******
******删减后地点数据大小：720******
[ 43.]
第0行预测完成！
[ 7.]
第1行预测完成！
[ 4.]
第2行预测完成！
[ 12.]
第3行预测完成！
[ 20.]
第4行预测完成！
[ 59.]
第5行预测完成！
[ 224.]
第6行预测完成！
[ 1579.]
第7行预测完成！
[ 2272.]
第8行预测完成！
[ 1526.]
第9行预测完成！
[ 1149.]
第10行预测完成！
[ 4006.]
第11行预测完成！
[ 4468.]
第12行预测完成！
[ 1916.]
第13行预测完成！
[ 990.]
第14行预测完成！
[ 989.]
第15行预测完成！
[ 1112.]
第16

第400行预测完成！
[ 2757.]
第401行预测完成！
[ 2383.]
第402行预测完成！
[ 1482.]
第403行预测完成！
[ 1089.]
第404行预测完成！
[ 871.]
第405行预测完成！
[ 505.]
第406行预测完成！
[ 185.]
第407行预测完成！
[ 21.]
第408行预测完成！
[ 4.]
第409行预测完成！
[-2.]
第410行预测完成！
[ 1.]
第411行预测完成！
[-1.]
第412行预测完成！
[ 18.]
第413行预测完成！
[ 127.]
第414行预测完成！
[ 554.]
第415行预测完成！
[ 1146.]
第416行预测完成！
[ 1183.]
第417行预测完成！
[ 1048.]
第418行预测完成！
[ 2025.]
第419行预测完成！
[ 2627.]
第420行预测完成！
[ 1570.]
第421行预测完成！
[ 1129.]
第422行预测完成！
[ 1033.]
第423行预测完成！
[ 1097.]
第424行预测完成！
[ 1824.]
第425行预测完成！
[ 1840.]
第426行预测完成！
[ 1248.]
第427行预测完成！
[ 896.]
第428行预测完成！
[ 726.]
第429行预测完成！
[ 457.]
第430行预测完成！
[ 132.]
第431行预测完成！
[ 12.]
第432行预测完成！
[-1.]
第433行预测完成！
[-1.]
第434行预测完成！
[ 1.]
第435行预测完成！
[-2.]
第436行预测完成！
[ 14.]
第437行预测完成！
[ 104.]
第438行预测完成！
[ 486.]
第439行预测完成！
[ 1001.]
第440行预测完成！
[ 1238.]
第441行预测完成！
[ 1056.]
第442行预测完成！
[ 1876.]
第443行预测完成！
[ 2425.]
第444行预测完成！
[ 1607.]
第445行预测完成！
[ 1193.]
第446行预测完成！
[ 1120.]
第447行预测完成！
[ 1178.]
第448行预测完成！
[ 2051.]
第449行预测完成！
[ 1968.]
第450行预测完成！
[ 1309.]
第451行预测完成！
[ 924.]
第452

第114行预测完成！
[ 529.]
第115行预测完成！
[ 392.]
第116行预测完成！
[ 403.]
第117行预测完成！
[ 476.]
第118行预测完成！
[ 394.]
第119行预测完成！
[ 327.]
第120行预测完成！
[ 281.]
第121行预测完成！
[ 259.]
第122行预测完成！
[ 285.]
第123行预测完成！
[ 313.]
第124行预测完成！
[ 318.]
第125行预测完成！
[ 340.]
第126行预测完成！
[ 487.]
第127行预测完成！
[ 614.]
第128行预测完成！
[ 595.]
第129行预测完成！
[ 488.]
第130行预测完成！
[ 707.]
第131行预测完成！
[ 940.]
第132行预测完成！
[ 742.]
第133行预测完成！
[ 545.]
第134行预测完成！
[ 473.]
第135行预测完成！
[ 482.]
第136行预测完成！
[ 809.]
第137行预测完成！
[ 792.]
第138行预测完成！
[ 598.]
第139行预测完成！
[ 481.]
第140行预测完成！
[ 535.]
第141行预测完成！
[ 568.]
第142行预测完成！
[ 595.]
第143行预测完成！
[ 447.]
第144行预测完成！
[ 327.]
第145行预测完成！
[ 287.]
第146行预测完成！
[ 292.]
第147行预测完成！
[ 300.]
第148行预测完成！
[ 311.]
第149行预测完成！
[ 347.]
第150行预测完成！
[ 609.]
第151行预测完成！
[ 679.]
第152行预测完成！
[ 621.]
第153行预测完成！
[ 499.]
第154行预测完成！
[ 757.]
第155行预测完成！
[ 986.]
第156行预测完成！
[ 767.]
第157行预测完成！
[ 550.]
第158行预测完成！
[ 476.]
第159行预测完成！
[ 455.]
第160行预测完成！
[ 828.]
第161行预测完成！
[ 792.]
第162行预测完成！
[ 616.]
第163行预测完成！
[ 506.]
第164行预测完成！
[ 576.]
第165行预测完成！
[ 615.]
第166行预测完成！
[

第547行预测完成！
[ 449.]
第548行预测完成！
[ 444.]
第549行预测完成！
[ 488.]
第550行预测完成！
[ 344.]
第551行预测完成！
[ 254.]
第552行预测完成！
[ 221.]
第553行预测完成！
[ 211.]
第554行预测完成！
[ 210.]
第555行预测完成！
[ 213.]
第556行预测完成！
[ 240.]
第557行预测完成！
[ 325.]
第558行预测完成！
[ 521.]
第559行预测完成！
[ 676.]
第560行预测完成！
[ 553.]
第561行预测完成！
[ 442.]
第562行预测完成！
[ 762.]
第563行预测完成！
[ 1005.]
第564行预测完成！
[ 750.]
第565行预测完成！
[ 545.]
第566行预测完成！
[ 445.]
第567行预测完成！
[ 419.]
第568行预测完成！
[ 800.]
第569行预测完成！
[ 758.]
第570行预测完成！
[ 528.]
第571行预测完成！
[ 431.]
第572行预测完成！
[ 422.]
第573行预测完成！
[ 460.]
第574行预测完成！
[ 325.]
第575行预测完成！
[ 213.]
第576行预测完成！
[ 193.]
第577行预测完成！
[ 185.]
第578行预测完成！
[ 183.]
第579行预测完成！
[ 192.]
第580行预测完成！
[ 224.]
第581行预测完成！
[ 287.]
第582行预测完成！
[ 388.]
第583行预测完成！
[ 523.]
第584行预测完成！
[ 492.]
第585行预测完成！
[ 411.]
第586行预测完成！
[ 675.]
第587行预测完成！
[ 896.]
第588行预测完成！
[ 638.]
第589行预测完成！
[ 481.]
第590行预测完成！
[ 408.]
第591行预测完成！
[ 384.]
第592行预测完成！
[ 680.]
第593行预测完成！
[ 687.]
第594行预测完成！
[ 504.]
第595行预测完成！
[ 410.]
第596行预测完成！
[ 402.]
第597行预测完成！
[ 421.]
第598行预测完成！
[ 300.]
第599行预测完成！


第259行预测完成！
[ 939.]
第260行预测完成！
[ 881.]
第261行预测完成！
[ 893.]
第262行预测完成！
[ 332.]
第263行预测完成！
[ 69.]
第264行预测完成！
[ 16.]
第265行预测完成！
[ 2.]
第266行预测完成！
[ 2.]
第267行预测完成！
[-1.]
第268行预测完成！
[ 12.]
第269行预测完成！
[ 91.]
第270行预测完成！
[ 360.]
第271行预测完成！
[ 795.]
第272行预测完成！
[ 966.]
第273行预测完成！
[ 825.]
第274行预测完成！
[ 1452.]
第275行预测完成！
[ 1956.]
第276行预测完成！
[ 1293.]
第277行预测完成！
[ 937.]
第278行预测完成！
[ 944.]
第279行预测完成！
[ 955.]
第280行预测完成！
[ 1627.]
第281行预测完成！
[ 1667.]
第282行预测完成！
[ 1252.]
第283行预测完成！
[ 922.]
第284行预测完成！
[ 872.]
第285行预测完成！
[ 872.]
第286行预测完成！
[ 328.]
第287行预测完成！
[ 78.]
第288行预测完成！
[ 34.]
第289行预测完成！
[ 19.]
第290行预测完成！
[ 13.]
第291行预测完成！
[ 10.]
第292行预测完成！
[ 20.]
第293行预测完成！
[ 111.]
第294行预测完成！
[ 649.]
第295行预测完成！
[ 1232.]
第296行预测完成！
[ 1119.]
第297行预测完成！
[ 892.]
第298行预测完成！
[ 1880.]
第299行预测完成！
[ 2544.]
第300行预测完成！
[ 1545.]
第301行预测完成！
[ 1168.]
第302行预测完成！
[ 1098.]
第303行预测完成！
[ 1074.]
第304行预测完成！
[ 2166.]
第305行预测完成！
[ 2058.]
第306行预测完成！
[ 1408.]
第307行预测完成！
[ 1116.]
第308行预测完成！
[ 1109.]
第309行预测完成！
[ 1150.]
第310行预测完成！
[ 447.]
第311行预测完

第690行预测完成！
[ 1359.]
第691行预测完成！
[ 1087.]
第692行预测完成！
[ 1018.]
第693行预测完成！
[ 997.]
第694行预测完成！
[ 318.]
第695行预测完成！
[ 9.]
第696行预测完成！
[ 4.]
第697行预测完成！
[ 6.]
第698行预测完成！
[ 8.]
第699行预测完成！
[ 12.]
第700行预测完成！
[ 20.]
第701行预测完成！
[ 135.]
第702行预测完成！
[ 812.]
第703行预测完成！
[ 1497.]
第704行预测完成！
[ 1163.]
第705行预测完成！
[ 964.]
第706行预测完成！
[ 1945.]
第707行预测完成！
[ 2828.]
第708行预测完成！
[ 1614.]
第709行预测完成！
[ 1200.]
第710行预测完成！
[ 1075.]
第711行预测完成！
[ 1096.]
第712行预测完成！
[ 2100.]
第713行预测完成！
[ 1893.]
第714行预测完成！
[ 1277.]
第715行预测完成！
[ 1006.]
第716行预测完成！
[ 927.]
第717行预测完成！
[ 900.]
第718行预测完成！
[ 297.]
第719行预测完成！
******删减前地点数据大小：1440******
******删减后地点数据大小：720******
[ 80.]
第0行预测完成！
[ 39.]
第1行预测完成！
[ 24.]
第2行预测完成！
[ 25.]
第3行预测完成！
[ 27.]
第4行预测完成！
[ 29.]
第5行预测完成！
[ 84.]
第6行预测完成！
[ 310.]
第7行预测完成！
[ 589.]
第8行预测完成！
[ 428.]
第9行预测完成！
[ 322.]
第10行预测完成！
[ 630.]
第11行预测完成！
[ 981.]
第12行预测完成！
[ 610.]
第13行预测完成！
[ 467.]
第14行预测完成！
[ 386.]
第15行预测完成！
[ 361.]
第16行预测完成！
[ 751.]
第17行预测完成！
[ 794.]
第18行预测完成！
[ 548.]
第19行预测完成！
[ 421.]
第20行预测完成！
[ 414.]
第21行预测完成！
[

第415行预测完成！
[ 263.]
第416行预测完成！
[ 329.]
第417行预测完成！
[ 292.]
第418行预测完成！
[ 493.]
第419行预测完成！
[ 741.]
第420行预测完成！
[ 451.]
第421行预测完成！
[ 339.]
第422行预测完成！
[ 298.]
第423行预测完成！
[ 322.]
第424行预测完成！
[ 537.]
第425行预测完成！
[ 608.]
第426行预测完成！
[ 427.]
第427行预测完成！
[ 341.]
第428行预测完成！
[ 328.]
第429行预测完成！
[ 317.]
第430行预测完成！
[ 135.]
第431行预测完成！
[ 42.]
第432行预测完成！
[ 5.]
第433行预测完成！
[-5.]
第434行预测完成！
[-6.]
第435行预测完成！
[-3.]
第436行预测完成！
[ 1.]
第437行预测完成！
[ 18.]
第438行预测完成！
[ 100.]
第439行预测完成！
[ 241.]
第440行预测完成！
[ 320.]
第441行预测完成！
[ 281.]
第442行预测完成！
[ 487.]
第443行预测完成！
[ 713.]
第444行预测完成！
[ 477.]
第445行预测完成！
[ 342.]
第446行预测完成！
[ 291.]
第447行预测完成！
[ 306.]
第448行预测完成！
[ 547.]
第449行预测完成！
[ 626.]
第450行预测完成！
[ 432.]
第451行预测完成！
[ 318.]
第452行预测完成！
[ 292.]
第453行预测完成！
[ 284.]
第454行预测完成！
[ 124.]
第455行预测完成！
[ 40.]
第456行预测完成！
[ 14.]
第457行预测完成！
[ 9.]
第458行预测完成！
[ 7.]
第459行预测完成！
[ 9.]
第460行预测完成！
[ 18.]
第461行预测完成！
[ 40.]
第462行预测完成！
[ 166.]
第463行预测完成！
[ 424.]
第464行预测完成！
[ 386.]
第465行预测完成！
[ 306.]
第466行预测完成！
[ 570.]
第467行预测完成！
[ 869.]
第468行预测完成！
[ 56

第141行预测完成！
[ 295.]
第142行预测完成！
[ 89.]
第143行预测完成！
[ 15.]
第144行预测完成！
[ 2.]
第145行预测完成！
[ 2.]
第146行预测完成！
[ 5.]
第147行预测完成！
[ 8.]
第148行预测完成！
[ 19.]
第149行预测完成！
[ 53.]
第150行预测完成！
[ 270.]
第151行预测完成！
[ 543.]
第152行预测完成！
[ 748.]
第153行预测完成！
[ 621.]
第154行预测完成！
[ 802.]
第155行预测完成！
[ 722.]
第156行预测完成！
[ 612.]
第157行预测完成！
[ 546.]
第158行预测完成！
[ 773.]
第159行预测完成！
[ 631.]
第160行预测完成！
[ 1053.]
第161行预测完成！
[ 735.]
第162行预测完成！
[ 467.]
第163行预测完成！
[ 415.]
第164行预测完成！
[ 346.]
第165行预测完成！
[ 256.]
第166行预测完成！
[ 62.]
第167行预测完成！
[ 3.]
第168行预测完成！
[-5.]
第169行预测完成！
[ 3.]
第170行预测完成！
[ 4.]
第171行预测完成！
[ 6.]
第172行预测完成！
[ 22.]
第173行预测完成！
[ 71.]
第174行预测完成！
[ 290.]
第175行预测完成！
[ 565.]
第176行预测完成！
[ 709.]
第177行预测完成！
[ 628.]
第178行预测完成！
[ 779.]
第179行预测完成！
[ 715.]
第180行预测完成！
[ 588.]
第181行预测完成！
[ 570.]
第182行预测完成！
[ 726.]
第183行预测完成！
[ 628.]
第184行预测完成！
[ 933.]
第185行预测完成！
[ 680.]
第186行预测完成！
[ 447.]
第187行预测完成！
[ 372.]
第188行预测完成！
[ 314.]
第189行预测完成！
[ 220.]
第190行预测完成！
[ 57.]
第191行预测完成！
[ 10.]
第192行预测完成！
[-1.]
第193行预测完成！
[ 1.]
第194行预测完成！
[ 4.]
第195行预

第591行预测完成！
[ 336.]
第592行预测完成！
[ 436.]
第593行预测完成！
[ 433.]
第594行预测完成！
[ 333.]
第595行预测完成！
[ 274.]
第596行预测完成！
[ 222.]
第597行预测完成！
[ 158.]
第598行预测完成！
[ 49.]
第599行预测完成！
[ 5.]
第600行预测完成！
[-7.]
第601行预测完成！
[-7.]
第602行预测完成！
[-5.]
第603行预测完成！
[-3.]
第604行预测完成！
[ 2.]
第605行预测完成！
[ 14.]
第606行预测完成！
[ 95.]
第607行预测完成！
[ 218.]
第608行预测完成！
[ 372.]
第609行预测完成！
[ 381.]
第610行预测完成！
[ 445.]
第611行预测完成！
[ 461.]
第612行预测完成！
[ 357.]
第613行预测完成！
[ 335.]
第614行预测完成！
[ 380.]
第615行预测完成！
[ 371.]
第616行预测完成！
[ 575.]
第617行预测完成！
[ 493.]
第618行预测完成！
[ 365.]
第619行预测完成！
[ 279.]
第620行预测完成！
[ 232.]
第621行预测完成！
[ 166.]
第622行预测完成！
[ 52.]
第623行预测完成！
[ 8.]
第624行预测完成！
[ 1.]
第625行预测完成！
[ 1.]
第626行预测完成！
[ 4.]
第627行预测完成！
[ 8.]
第628行预测完成！
[ 15.]
第629行预测完成！
[ 41.]
第630行预测完成！
[ 156.]
第631行预测完成！
[ 412.]
第632行预测完成！
[ 596.]
第633行预测完成！
[ 501.]
第634行预测完成！
[ 647.]
第635行预测完成！
[ 673.]
第636行预测完成！
[ 512.]
第637行预测完成！
[ 484.]
第638行预测完成！
[ 607.]
第639行预测完成！
[ 558.]
第640行预测完成！
[ 952.]
第641行预测完成！
[ 680.]
第642行预测完成！
[ 461.]
第643行预测完成！
[ 357.]
第644行预测完成！
[ 299.]
第6

In [44]:
nov_data.sort_values(by=['time_stamp','loc_id'],inplace=True)

In [47]:
# 预测的负值处理
nov_data.loc[nov_data.num_of_people<0] = 0

In [45]:
nov_data.iloc[:,:3].to_csv('../datasets/nov_prediction.csv',index=False)

## 4. 线上记录
* 3-9月 前一个非null值填充 去掉二月数据 异常值处理 训练xgb1000轮 线下rmse：377 线上rmse：355
* 3-10月 前一个非null值填充 去掉二月数据 异常值处理 训练xgb2000轮 线上rmse：237
* 2-10月 0填充 增加小时特征 训练xgb2000轮 线上rmse：241
* 4-10月 前一个非null值填充 异常值处理 去掉十一 增加小时特征 训练xgb2000轮 线上rmse：

# 公共函数

In [6]:
# 按地点分组
def locID_split(data):
    res = []
    df = data.copy()
    for i in df.loc_id.unique():
        tmp = df[df.loc_id == i]
        tmp.reset_index(inplace=True)
        tmp.drop('index',inplace=True,axis=1)
        res.append(tmp)
    return res

In [14]:
# 实时预测填充
def real_time_predict(model,data):
    df = data.copy()
    test_list = locID_split(df)
    res = pd.DataFrame()
    for tmp in test_list:
        for i in range(tmp.shape[0]):
            X = xgb.DMatrix(tmp.iloc[i:i+1,3:])
            pred = np.round(model.predict(X,ntree_limit = model.best_iteration))
            print(pred)
            tmp.loc[i,'num_of_people'] = pred
            for j in range(len(columns_24h)):
                tmp.loc[i+j+1,columns_24h[j]] = pred
            for k in range(len(columns_30d)):
                tmp.loc[i+(k+1)*24,columns_30d[k]] = pred
            print('第%s行预测完成！'%i)
        print('******删减前地点数据大小：%s******'%tmp.shape[0])
        tmp.dropna(axis=0,inplace=True)
        print('******删减后地点数据大小：%s******'%tmp.shape[0])
        res = res.append(tmp)
    return res

In [8]:
reg = re.compile('\s+|\t|\n') # 正则匹配空格、Tab键、换行
#列名，历史24小时人数
col_24h='''history_1h_population,history_2h_population,history_3h_population,history_4h_population
    ,history_5h_population,history_6h_population,history_7h_population,history_8h_population
    ,history_9h_population,history_10h_population,history_11h_population,history_12h_population
    ,history_13h_population,history_14h_population,history_15h_population,history_16h_population,history_17h_population
    ,history_18h_population,history_19h_population,history_20h_population,history_21h_population,history_22h_population
    ,history_23h_population'''
columns_24h=re.sub(reg,'',col_24h).split(',')  
#列名，同一时刻历史15天人数
col_30d='''history_1d_population,history_2d_population,history_3d_population,
    history_4d_population,history_5d_population,history_6d_population,
    history_7d_population,history_8d_population,history_9d_population,history_10d_population,history_11d_population,
    history_12d_population,history_13d_population,history_14d_population,history_15d_population,history_16d_population
    ,history_17d_population,history_18d_population,history_19d_population,history_20d_population,history_21d_population
    ,history_22d_population,history_23d_population,history_24d_population,history_25d_population,history_26d_population
    ,history_27d_population,history_28d_population,history_29d_population,history_30d_population'''
columns_30d=re.sub(reg,'',col_30d).split(',')  

In [46]:
def xgboost_model_train(train_X,train_y,test_X,test_y):
    # 模型训练
    xgb_train  = xgb.DMatrix(train_X, train_y)
    xgb_test   = xgb.DMatrix(test_X, test_y)
    params_xgb = {
#         'booster': 'gbtree',                # 取决于使用哪种booster,有两中模型可以选择gbtree和gblinear。gbtree使用基于树的模型进行提升计算，gblinear使用线性模型进行提升计算
        'objective': 'reg:linear',     # 定义学习任务及相应的学习目标，可选的目标函数如下：“reg:linear” —— 线性回归,“reg:logistic”—— 逻辑回归。
        #'multi:softmax'
        'gamma': 0.03,                      # 最小损失减少，模型在默认情况下，对于一个节点的划分只有在其loss function 得到结果大于0的情况下才进行，而gamma 给定了所需的最低loss function的值
        'max_depth': 8,                     # 数的最大深度。缺省值为6，树的深度越大，则对数据的拟合程度越高（过拟合程度也越高）。即该参数也是控制过拟合
        'lambda': 4,                        # L2 正则的惩罚系数
        'subsample': 0.8,                   # 用于训练模型的子样本占整个样本集合的比例。如果设置为0.5则意味着XGBoost将随机的从整个样本集合中随机的抽取出50%的子样本建立树模型，这能够防止过拟合。
        'colsample_bytree': 0.8,            # 在建立树时对特征采样的比例。缺省值为1。
#         'eval_metric':'mape_ln',
        'min_child_weight': 8.5,            # 孩子节点中最小的样本权重和，在现行回归模型中，这个参数是指建立每个模型所需要的最小样本数。该成熟越大算法越conservative。即调大这个参数能够控制过拟合。
        'silent': 1,                        # 取0时表示打印出运行时信息，取1时表示以缄默方式运行，不打印运行时信息。缺省值为0。
        'eta': 0.03,                        # 为了防止过拟合，更新过程中用到的收缩步长。通常最后设置eta为0.01~0.2
        'seed': 1000}                       # 随机数的种子。缺省值为0。
    num_boost_round = 2000
    plst  = params_xgb.items()
    model_xgb = xgb.train(plst, xgb_train, num_boost_round,evals=[(xgb_train,'train'),(xgb_test,'test')],early_stopping_rounds=20,verbose_eval=True)
    return model_xgb